difforig config/inter/

2008-03-19 14:51:18  <rurban@x-ray.at>

diff -ub  config/inter/make.pm.orig
--- config/inter/make.pm.orig	2008-03-15 23:01:20.000000000 +0100
+++ config/inter/make.pm	2008-03-19 14:50:29.218750000 +0100
@@ -42,10 +42,10 @@
 
     # check the candidates for a 'make' program in this order:
     # environment ; option ; probe ; ask ; default
-    # first pick wins.
+    # first pick wins. On cygwin prefer make over nmake.
     $prog ||= $ENV{ uc($util) };
     $prog ||= $conf->options->get($util);
-    $prog ||= check_progs( ['gmake', 'mingw32-make', 'nmake', 'make'], $verbose );
+    $prog ||= check_progs( $^O eq 'cygwin' ? ['gmake', 'make'] : ['gmake', 'mingw32-make', 'nmake', 'make'], $verbose );
     if ( !$prog ) {
         $prog = ( $conf->options->get('ask') )
             ? prompt( $prompt, $prog ? $prog : $conf->data->get($util) )
