difforig config/gen/makefiles/root.in

2008-03-20 12:53:29  <rurban@x-ray.at>

diff -ub config/gen/makefiles/root.in.orig config/gen/makefiles/root.in
--- config/gen/makefiles/root.in.orig	2008-03-18 00:15:14.000000000 +0100
+++ config/gen/makefiles/root.in	2008-03-20 12:52:47.234375000 +0100
@@ -784,6 +784,7 @@
 	$(PARROT) -o $@ runtime/parrot/library/parrotlib.pir
 
 runtime/parrot/include/config.fpmc : myconfig config_lib.pasm $(MINIPARROT)
+	-test -e $@ -a -s $@ || ( echo "Removing empty $@"; rm $@ )
 	@echo Invoking Parrot to generate runtime/parrot/include/config.fpmc --cross your fingers
 	$(MINIPARROT) config_lib.pasm > $@
 
