diffbk PLATFORMS

diff -u PLATFORMS~ ./PLATFORMS
--- PLATFORMS~	2008-05-13 20:35:22.359375000 +0200
+++ ./PLATFORMS	2008-05-20 19:05:56.421875000 +0200
@@ -41,6 +41,7 @@
 opensolaris-x86-gcc_4.0.3     4   Y    Y    ?     ?   ?   Y    Y/2   ?  20080325
 tru64-alpha-compaq_c6.5       8   ?    ?    ?     Y   ?   Y    Y/2   ?  20060203
 cygwin1.5.24-x86-gcc3.4.4     4   Y    Y    -     -   -   Y    Y/6*1 ?  20071016
+cygwin1.5.25-x86-gcc3.4.4     4   Y    Y    -     -   -   Y    Y/6*1 ?  20080513
 win32-x86-mingw_gcc3.4.2          Y    Y    -     -   -   Y    Y/10  Y  20080415
 win32-x86-mingw_gcc4.2.1          Y    Y    -     -   -   Y    Y/10  ?  20080318
 win32-x86-msvc_6.0                -    Y    -     Y   -   Y    Y/6   ?  20080318
