diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/clamav-0.90/clamav-config.h.in src/clamav-0.90/clamav-config.h.in
--- origsrc/clamav-0.90/clamav-config.h.in	2007-02-13 01:31:24.000000000 +0000
+++ src/clamav-0.90/clamav-config.h.in	2007-02-22 08:09:02.328125000 +0000
@@ -78,6 +78,9 @@
 /* "default FD_SETSIZE value" */
 #undef DEFAULT_FD_SETSIZE
 
+/* Define if your struct dirent doesn't contain d_ino */
+#undef DIRENT_MISSING_D_INO
+
 /* file i/o buffer size */
 #undef FILEBUFF
 
@@ -90,6 +93,9 @@
 /* access rights in msghdr */
 #undef HAVE_ACCRIGHTS_IN_MSGHDR
 
+/* Define to 1 if you have the <arpa/nameser.h> header file. */
+#undef HAVE_ARPA_NAMESER_H
+
 /* attrib aligned */
 #undef HAVE_ATTRIB_ALIGNED
 
@@ -177,6 +183,12 @@
 /* Define to 1 if you have the <ndir.h> header file. */
 #undef HAVE_NDIR_H
 
+/* Define to 1 if you have the <netdb.h> header file. */
+#undef HAVE_NETDB_H
+
+/* Define to 1 if you have the <netinet/in.h> header file. */
+#undef HAVE_NETINET_IN_H
+
 /* Define to 1 if you have the `poll' function. */
 #undef HAVE_POLL
 
@@ -198,7 +210,7 @@
 /* Define to 1 if you have the <regex.h> header file. */
 #undef HAVE_REGEX_H
 
-/* have resolv.h */
+/* Define to 1 if you have the <resolv.h> header file. */
 #undef HAVE_RESOLV_H
 
 /* Define to 1 if you have the `sendmsg' function. */
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/clamav-0.90/clamscan/treewalk.c src/clamav-0.90/clamscan/treewalk.c
--- origsrc/clamav-0.90/clamscan/treewalk.c	2007-01-30 20:50:33.000000000 +0000
+++ src/clamav-0.90/clamscan/treewalk.c	2007-02-19 21:02:59.144000000 +0000
@@ -99,7 +99,7 @@
 
     if((dd = opendir(dirname)) != NULL) {
 	while((dent = readdir(dd))) {
-#ifndef C_INTERIX
+#ifndef DIRENT_MISSING_D_INO
 	    if(dent->d_ino)
 #endif
 	    {
@@ -196,7 +196,7 @@
 
     if((dd = opendir(dirname)) != NULL) {
 	while((dent = readdir(dd))) {
-#ifndef C_INTERIX
+#ifndef DIRENT_MISSING_D_INO
 	    if(dent->d_ino)
 #endif
 	    {
@@ -242,7 +242,7 @@
 
     if((dd = opendir(dirname)) != NULL) {
 	while((dent = readdir(dd))) {
-#ifndef C_INTERIX
+#ifndef DIRENT_MISSING_D_INO
 	    if(dent->d_ino)
 #endif
 	    {
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/clamav-0.90/configure.in src/clamav-0.90/configure.in
--- origsrc/clamav-0.90/configure.in	2007-02-13 01:10:47.000000000 +0000
+++ src/clamav-0.90/configure.in	2007-02-22 07:57:02.078125000 +0000
@@ -58,6 +58,27 @@
 AC_FUNC_MMAP
 AC_FUNC_FSEEKO
 
+
+dnl Check for broken cygwin-1.5.19 without d_ino or INTERIX (code by David Rothenberger <daveroth*acm.org>)
+AC_MSG_CHECKING([whether struct dirent contains d_ino])
+AC_RUN_IFELSE(
+	[AC_LANG_SOURCE([[
+#include <sys/types.h>
+#include <dirent.h>
+int main(void){struct dirent d;exit(sizeof(d.d_ino)>=0);}
+	]])],
+	[AC_MSG_RESULT(yes)],
+	[
+		AC_MSG_RESULT(no)
+		AC_DEFINE(DIRENT_MISSING_D_INO, 1,
+			[Define if your struct dirent doesn't contain d_ino])
+	],
+	[
+		AC_MSG_WARN([cross compiling: assuming DIRENT_MISSING_D_INO])
+		AC_DEFINE(DIRENT_MISSING_D_INO)
+	]
+)
+
 dnl Check for broken snprintf (code by Phil Oleson <oz*nixil.net>)
 if test "x$ac_cv_func_snprintf" = "xyes" ; then
         AC_MSG_CHECKING([whether snprintf correctly terminates long strings])
@@ -166,7 +187,13 @@
     if test -z "$resolv_lib"; then
 	AC_CHECK_LIB(resolv, dn_expand, resolv_lib="-lresolv",)
     fi
-    AC_CHECK_HEADER(resolv.h,[FRESHCLAM_LIBS="$FRESHCLAM_LIBS $resolv_lib"; CLAMAV_MILTER_LIBS="$CLAMAV_MILTER_LIBS $resolv_lib"; AC_DEFINE(HAVE_RESOLV_H,1,have resolv.h)], AC_MSG_WARN([****** DNS support disabled]), [#include <netinet/in.h>])
+    AC_HEADER_RESOLV
+    if test $ac_cv_header_resolv_h = yes; then
+	FRESHCLAM_LIBS="$FRESHCLAM_LIBS $resolv_lib"
+	CLAMAV_MILTER_LIBS="$CLAMAV_MILTER_LIBS $resolv_lib"
+    else
+	AC_MSG_WARN([****** DNS support disabled])
+    fi
 fi
 
 AC_ARG_ENABLE(clamuko,
@@ -423,6 +450,8 @@
 	AC_DEFINE(CL_THREAD_SAFE,1,[thread safe])
 	AC_DEFINE(_REENTRANT,1,[thread safe])
     fi
+    AC_DEFINE(BUILD_CLAMD, 1, "build clamd")
+    AM_CONDITIONAL(BUILD_CLAMD, 1)
     use_gethostbyname_r="no"
     ;;
 solaris*)
@@ -579,6 +608,8 @@
 	AC_DEFINE(CL_THREAD_SAFE,1,[thread safe])
 	AC_DEFINE(_REENTRANT,1,[thread safe])
     fi
+    AC_DEFINE(BUILD_CLAMD, 1, "build clamd")
+    AM_CONDITIONAL(BUILD_CLAMD, 1)
     ;;
 *)
     ;;
@@ -618,7 +649,7 @@
     ])
 fi
 
-AM_CONDITIONAL(BUILD_CLAMD, test "$have_pthreads" = "yes")
+AM_CONDITIONAL(BUILD_CLAMD, test "$have_pthreads" = "yes" -o "$target_os" = "cygwin")
 AM_CONDITIONAL(HAVE_MILTER, test "$have_milter" = "yes")
 
 if test "$have_pthreads" = "yes"
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/clamav-0.90/contrib/init/Cygwin/clamd src/clamav-0.90/contrib/init/Cygwin/clamd
--- origsrc/clamav-0.90/contrib/init/Cygwin/clamd	1970-01-01 00:00:00.000000000 +0000
+++ src/clamav-0.90/contrib/init/Cygwin/clamd	2007-02-19 21:02:59.440875000 +0000
@@ -0,0 +1,152 @@
+#! /bin/bash
+#
+# crond   Start/Stop the clam antivirus daemon. 0.8x series
+# v1.0 2004-10-18 Reini Urban rurban@x-ray.at
+#
+# chkconfig: 2345 90 60
+# description: clamd is a standard UNIX program that scans for Viruses.
+# processname: clamd
+# config: /etc/clamd.conf
+# pidfile: /var/run/clamd.pid
+
+PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
+PREFIX=/usr
+CONFIG=/etc/clamd.conf
+PIDFILE=/var/run/clamd.pid
+LOGFILE=/var/log/clamd.log
+SOCKET=/var/run/clamd.sock
+
+# Source function library.
+if [ -f /etc/init.d/functions ] ; then
+  . /etc/init.d/functions
+fi
+
+RETVAL=0
+
+# See how we were called.
+
+prog="clamd"
+progdir="/usr/sbin"
+DAEMON="$progdir/$prog.exe"
+
+test -f $DAEMON || exit 0
+
+# Source configuration
+if [ -f /etc/sysconfig/$prog ] ; then
+    . /etc/sysconfig/$prog
+fi
+
+start() {
+        echo -n $"Starting $prog: "
+	# check if cygrunsrv process
+        cygrunsrv --start $prog
+        RETVAL=$?
+        echo
+        [ $RETVAL -eq 0 ] && touch $PIDFILE  && echo "done."
+        return $RETVAL
+}
+
+stop() {
+        echo -n $"Stopping $prog: "
+	cygrunsrv --stop $prog
+        if test -r $PIDFILE; then
+            kill `cat $PIDFILE` 2> /dev/null || echo -n " failed"
+        else
+            echo -n " no PID file"
+        fi
+	# this really needs a long time to stop.
+        RETVAL=$?
+        echo "."
+        [ $RETVAL -eq 0 ] && rm -f $PIDFILE $SOCKET && echo "done."
+        return $RETVAL
+}
+
+rhstatus() {
+        # service status
+        cygrunsrv -Q $prog
+        # connect(): No such file or directory
+	/usr/bin/clamdscan /usr/sbin/clamd.exe
+	#status clamd
+}
+
+restart() {
+        echo -n $"Restarting $prog: "
+        $0 stop
+        sleep 1
+        $0 start
+        echo "done."
+}
+
+installcron() {
+    ( crontab -l | grep freshclam >/dev/null 2>&1 ) || ( \
+	echo "add something like this to your cronjob:"
+	echo
+	echo "*  */3 * * *	/usr/bin/freshclam >>/var/log/freshclam.log 2>&1"
+	echo "#*  */3 * * *	/usr/bin/freshclam >>/var/log/freshclam.log 2>&1" >> /var/cron/tabs/$USER
+	kill -s HUP `cat /var/run/cron.pid`
+	crontab -e )
+}
+install() {
+        freshclam
+        echo -n $"Installing $prog daemon: "
+	# some safety measures
+	touch $LOGFILE $PIDFILE
+	chgrp 18 $LOGFILE $PIDFILE
+	chmod g+w $LOGFILE $PIDFILE
+	rm -f $SOCKET
+	#it was compiled with uid=18
+	cygrunsrv --install $prog --path $DAEMON -e CYGWIN="$CYGWIN" --disp "CYGWIN `$DAEMON --version`"
+        echo "done."
+}
+uninstall() {
+        echo -n $"Uninstalling $prog daemon: "
+	stop
+	cygrunsrv --remove $prog
+        echo "done."
+}
+reload() {
+        echo -n $"Reloading $prog daemon configuration: "
+	echo "unsupported. $0 restart"
+	return 1
+
+	/usr/bin/kill -HUP `cat $PIDFILE`
+        RETVAL=$?
+        [ $RETVAL -eq 0 ] && echo "done."
+        RETVAL=$?
+        return $RETVAL
+}
+
+case "$1" in
+  start)
+        start
+        ;;
+  stop)
+        stop
+        ;;
+  restart)
+        restart
+        ;;
+  reload)
+        reload
+        ;;
+  install)
+        install
+        ;;
+  installcron)
+        installcron
+        ;;
+  uninstall)
+        uninstall
+        ;;
+  status)
+        rhstatus
+        ;;
+  condrestart)
+        [ -f $PIDFILE ] && restart || :
+        ;;
+  *)
+        echo $"Usage: $0 {start|stop|status|install|uninstall|installcron|restart|condrestart}"
+        exit 1
+esac
+
+exit $?
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/clamav-0.90/etc/clamd.conf src/clamav-0.90/etc/clamd.conf
--- origsrc/clamav-0.90/etc/clamd.conf	2007-02-12 18:18:42.000000000 +0000
+++ src/clamav-0.90/etc/clamd.conf	2007-02-19 21:02:59.565875000 +0000
@@ -5,7 +5,7 @@
 
 
 # Comment or remove the line below.
-Example
+#Example
 
 # Uncomment this option to enable logging.
 # LogFile must be writable for the user running daemon.
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/clamav-0.90/etc/freshclam.conf src/clamav-0.90/etc/freshclam.conf
--- origsrc/clamav-0.90/etc/freshclam.conf	2007-02-11 09:54:46.000000000 +0000
+++ src/clamav-0.90/etc/freshclam.conf	2007-02-19 21:02:59.769000000 +0000
@@ -5,7 +5,7 @@
 
 
 # Comment or remove the line below.
-Example
+#Example
 
 # Path to the database directory.
 # WARNING: It must match clamd.conf's directive!
@@ -52,7 +52,7 @@
 
 # Uncomment the following line and replace XY with your country
 # code. See http://www.iana.org/cctld/cctld-whois.htm for the full list.
-#DatabaseMirror db.XY.clamav.net
+DatabaseMirror db.de.clamav.net
 
 # database.clamav.net is a round-robin record which points to our most 
 # reliable mirrors. It's used as a fall back in case db.XY.clamav.net is 
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/clamav-0.90/sigtool/sigtool.c src/clamav-0.90/sigtool/sigtool.c
--- origsrc/clamav-0.90/sigtool/sigtool.c	2007-02-12 20:55:24.000000000 +0000
+++ src/clamav-0.90/sigtool/sigtool.c	2007-02-19 21:03:22.972125000 +0000
@@ -1018,7 +1018,7 @@
     }
 
     while((dent = readdir(dd))) {
-#ifndef C_INTERIX
+#ifndef DIRENT_MISSING_D_INO
 	if(dent->d_ino)
 #endif
 	{
@@ -1551,7 +1551,7 @@
     }
 
     while((dent = readdir(dd))) {
-#ifndef C_INTERIX
+#ifndef DIRENT_MISSING_D_INO
 	if(dent->d_ino)
 #endif
 	{
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/clamav-0.90/sigtool/vba.c src/clamav-0.90/sigtool/vba.c
--- origsrc/clamav-0.90/sigtool/vba.c	2007-02-11 00:23:58.000000000 +0000
+++ src/clamav-0.90/sigtool/vba.c	2007-02-19 21:03:00.065875000 +0000
@@ -963,7 +963,10 @@
 
     if ((dd = opendir (dirname)) != NULL) {
 	while ((dent = readdir (dd))) {
-	    if (dent->d_ino) {
+#ifndef DIRENT_MISSING_D_INO
+	    if (dent->d_ino) 
+#endif
+	    {
 		if (strcmp (dent->d_name, ".") && strcmp (dent->d_name, "..")) {
 		    /* build the full name */
 		    fname = (char *) cli_calloc (strlen (dirname) + strlen (dent->d_name) + 2, sizeof (char));
@@ -1120,7 +1123,10 @@
 
     if ((dd = opendir (dirname)) != NULL) {
 	while ((dent = readdir (dd))) {
-	    if (dent->d_ino) {
+#ifndef DIRENT_MISSING_D_INO
+	    if (dent->d_ino) 
+#endif
+	    {
 		if (strcmp (dent->d_name, ".") && strcmp (dent->d_name, "..")) {
 		    /* build the full name */
 		    fname = calloc (strlen (dirname) + strlen (dent->d_name) + 2, sizeof (char));
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/clamav-0.90/test/negative.expected src/clamav-0.90/test/negative.expected
--- origsrc/clamav-0.90/test/negative.expected	1970-01-01 00:00:00.000000000 +0000
+++ src/clamav-0.90/test/negative.expected	2007-02-19 21:03:00.097125000 +0000
@@ -0,0 +1,6 @@
+clam-error.rar: RAR module failure
+clam.cab: ClamAV-Test-File FOUND
+clam.exe: ClamAV-Test-File FOUND
+clam.exe.bz2: ClamAV-Test-File FOUND
+clam.rar: ClamAV-Test-File FOUND
+clam.zip: ClamAV-Test-File FOUND
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/clamav-0.90/test/positive.expected src/clamav-0.90/test/positive.expected
--- origsrc/clamav-0.90/test/positive.expected	1970-01-01 00:00:00.000000000 +0000
+++ src/clamav-0.90/test/positive.expected	2007-02-19 21:03:00.362750000 +0000
@@ -0,0 +1,7 @@
+test: ClamAV-Test-Signature FOUND
+test-failure.rar: RAR module failure
+test-zip-noext: ClamAV-Test-Signature FOUND
+test.bz2: ClamAV-Test-Signature FOUND
+test.msc: ClamAV-Test-Signature FOUND
+test.rar: ClamAV-Test-Signature FOUND
+test.zip: ClamAV-Test-Signature FOUND
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/clamav-0.90/test/positive.result src/clamav-0.90/test/positive.result
--- origsrc/clamav-0.90/test/positive.result	1970-01-01 00:00:00.000000000 +0000
+++ src/clamav-0.90/test/positive.result	2007-02-22 08:16:04.406250000 +0000
@@ -0,0 +1 @@
+test: ClamAV-Test-Signature FOUND
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/clamav-0.90/test/test src/clamav-0.90/test/test
--- origsrc/clamav-0.90/test/test	1970-01-01 00:00:00.000000000 +0000
+++ src/clamav-0.90/test/test	2007-02-19 21:03:00.440875000 +0000
@@ -0,0 +1 @@
+$CEliacmaTrESTuScikgsn$FREE-TEST-SIGNATURE$EEEEE$
