--- origsrc/icu/source/common/unicode/pwin32.h	2011-07-19 23:16:18.000000000 +0200
+++ src/icu/source/common/unicode/pwin32.h	2011-07-26 12:45:13.284604700 +0200
@@ -46,6 +46,11 @@
 #       define U_HAVE_INTTYPES_H 0
 #   endif
 #endif
+/* allow #include <windows.h> */ 
+#ifdef __CYGWIN__
+#   define U_HAVE_INTTYPES_H 1
+#endif
+
 
 /**
  * Define what support for C++ streams is available.
--- origsrc/icu/source/common/unicode/rbbi.h	2011-07-19 23:16:18.000000000 +0200
+++ src/icu/source/common/unicode/rbbi.h	2011-07-26 12:45:13.293603800 +0200
@@ -663,12 +663,14 @@ protected:
     virtual int32_t getBreakType() const;
 #endif
 
+public:
     /**
       * Set the type of the break iterator.
       * @internal
       */
     virtual void setBreakType(int32_t type);
 
+protected:
     /**
       * Common initialization function, used by constructors and bufferClone.
       *   (Also used by DictionaryBasedBreakIterator::createBufferClone().)
