--- origsrc/icu/source/CYGWIN-PATCHES/README	1970-01-01 01:00:00.000000000 +0100
+++ src/icu/source/CYGWIN-PATCHES/README	2010-09-16 09:51:21.525348500 +0200
@@ -0,0 +1,155 @@
+icu
+-------------------------------------------
+IBM Internationalization Components for Unicode
+
+Runtime requirements:
+  cygwin-1.7
+  libgcc1-4.3.4
+  libstdc++6-4.3.4
+
+Build requirements:
+  cygwin-1.7
+  autoconf2.5-2.61
+  binutils
+  cygport
+  gawk
+  gcc-4.3.4
+  make
+  sed
+
+Canonical homepage:
+  http://ibm.com/software/globalization/icu/
+
+Canonical download:
+  ftp://www-126.ibm.com/pub/icu/
+
+-------------------------------------------
+
+Build instructions:
+  unpack icu-4.5.1-X-src.tar.bz2
+    if you use setup to install this src package, it will be
+	 unpacked under /usr/src automatically
+  cd /usr/src
+  cygport ./icu-4.5.1-X all
+
+This will create:
+  /usr/src/icu-4.5.1-X-src.tar.bz2
+  /usr/src/icu-4.5.1-X.tar.bz2
+  /usr/src/libicu38-4.5.1-X.tar.bz2
+  /usr/src/libicu-devel-4.5.1-X.tar.bz2
+  /usr/src/icu-doc-4.5.1-X.tar.bz2
+
+-------------------------------------------
+
+Files included in the binary packages:
+
+=icu= package:
+
+  usr/share/doc/Cygwin/icu-4.5.1.README
+
+=libicu45= package:
+
+  usr/bin/cygicudt45.dll	The huge DATA library
+  usr/bin/cygicuin45.dll	i18n Library
+  usr/bin/cygicuio45.dll	Optional stdio-like API with Unicode support
+  usr/bin/cygicule45.dll	Optional Font Layout Engine  
+  usr/bin/cygiculx45.dll 	Optional Font Layout Extensions Engine
+  usr/bin/cygicuuc45.dll	Base library required by all other ICU libraries
+
+=libicu-devel= package:
+
+  usr/bin/cygicuut45.dll	Internal Tool Utility Library
+  usr/bin/derb.exe
+  usr/bin/genbrk.exe
+  usr/bin/gencnval.exe
+  usr/bin/genctd.exe
+  usr/bin/genrb.exe
+  usr/bin/icu-config
+  usr/bin/makeconv.exe
+  usr/bin/pkgdata.exe
+  usr/bin/uconv.exe
+  usr/share/man/man1/derb.1
+  usr/share/man/man1/genbrk.1
+  usr/share/man/man1/gencnval.1
+  usr/share/man/man1/genctd.1
+  usr/share/man/man1/genrb.1
+  usr/share/man/man1/icu-config.1
+  usr/share/man/man1/makeconv.1
+  usr/share/man/man1/pkgdata.1
+  usr/share/man/man1/uconv.1
+  usr/share/man/man8/genccode.8
+  usr/share/man/man8/gencmn.8
+  usr/share/man/man8/gensprep.8
+  usr/share/man/man8/genuca.8
+  usr/share/man/man8/icupkg.8
+  usr/sbin/genccode.exe
+  usr/sbin/gencmn.exe
+  usr/sbin/gensprep.exe
+  usr/sbin/genuca.exe
+  usr/sbin/icupkg.exe
+  usr/sbin/icuswap.exe
+  usr/include/layout/*
+  usr/include/unicode/*
+  usr/lib/icu/4.5/Makefile.inc
+  usr/lib/icu/Makefile.inc
+  usr/lib/icu/current
+  usr/lib/libicudt45.0.dll.a
+  usr/lib/libicudt45.dll.a
+  usr/lib/libicuin45.0.dll.a
+  usr/lib/libicuin45.dll.a
+  usr/lib/libicuio45.0.dll.a
+  usr/lib/libicuio45.dll.a
+  usr/lib/libicule45.0.dll.a
+  usr/lib/libicule45.dll.a
+  usr/lib/libiculx45.0.dll.a
+  usr/lib/libiculx45.dll.a
+  usr/lib/libicuuc45.0.dll.a
+  usr/lib/libicuuc45.dll.a
+  usr/share/icu/4.5/config/mh-cygwin
+  usr/share/icu/4.5/mkinstalldirs
+
+=icu-doc= package:
+
+  usr/share/doc/icu/license.html
+  usr/share/doc/icu/PACKAGES 
+  usr/share/doc/icu/readme.html 
+  usr/share/doc/icu/unicode-license.txt   
+  usr/share/doc/icu/html/*.html
+
+------------------
+
+Port Notes:
+
+----- version 4.5.1-1 -----
+most patches applied. at least 4.2 needed for hiphop.
+
+----- version 3.8-7 -----
+recompile with latest gcc-4.3.4
+updates dependency
+
+----- version 3.8-6 -----
+release-2: fixed packaging: patches
+
+----- version 3.8-5 -----
+release-2 test package with gcc-4.3
+
+----- version 3.8-4 -----
+Allow #include <windows.h> (patched unicode/pwin32.h)
+
+----- version 3.8-3 -----
+Official cygwin release. Merge of Yaakov's 
+version with mine: Yaakov's simplier cygport, 
+Reini's layout + doc fixes (reini)
+
+----- version 3.8-2 -----
+Patch from Gentoo Linux for CVE-2007-4770,4771. (yaakov)
+
+----- version 3.8-1 -----
+Long-overdue version bump. (yaakov)
+Another version was ITP'd at cygwin. (reini)
+
+----- version 3.2-1 -----
+Initial release for Cygwin. (yaakov)
+
+Cygwin port maintained by:
+Reini Urban
--- origsrc/icu/source/CYGWIN-PATCHES/devel.hint	1970-01-01 01:00:00.000000000 +0100
+++ src/icu/source/CYGWIN-PATCHES/devel.hint	2010-09-16 09:34:46.405846500 +0200
@@ -0,0 +1,17 @@
+category: Devel
+requires: icu libicu38 gcc4-core gcc4-g++
+external-source: icu
+sdesc: "IBM Internationalization Components for Unicode (development)"
+ldesc: "All the files necessary to write applications that use ICU, 
+along with examples and some documentation.
+- The /usr/include/unicode directory which contains all the ICU
+headers.
+- The libraries and import libraries, versioned and unversioned.
+- A sample Makefile fragment that can be included by applications
+using ICU, to faciliate their building, along with a platform-specific
+configuration file included by this fragment.
+- The sample applications from the ICU source tree, in an appropriate
+location for the system that the package is installed on (for example,
+in /usr/share/doc/icu-4.5.1-5/examples).
+This package depends on the ICU libraries package with the exact same
+version, since it provides .dll.a symbolic links to the latest libraries."
--- origsrc/icu/source/CYGWIN-PATCHES/doc.hint	1970-01-01 01:00:00.000000000 +0100
+++ src/icu/source/CYGWIN-PATCHES/doc.hint	2010-09-16 09:34:46.416847600 +0200
@@ -0,0 +1,5 @@
+sdesc: "The ICU docs package"
+ldesc: "The doxygen generated html set for icu." 
+category: Devel Doc
+requires: 
+external-source: icu
--- origsrc/icu/source/CYGWIN-PATCHES/icu.hint	1970-01-01 01:00:00.000000000 +0100
+++ src/icu/source/CYGWIN-PATCHES/icu.hint	2010-09-16 09:34:46.428848800 +0200
@@ -0,0 +1,10 @@
+category: Libs
+requires: cygwin libicu38
+sdesc: "IBM Internationalization Components for Unicode (apps and docs)"
+ldesc: "ICU is a set of C and C++ libraries that provides robust and full-featured
+Unicode and locale support. The library provides calendar support, conversions
+for many character sets, language sensitive collation, date
+and time formatting, support for many locales, message catalogs
+and resources, message formatting, normalization, number and currency
+formatting, time zones support, transliteration, word, line and
+sentence breaking, etc."
--- origsrc/icu/source/CYGWIN-PATCHES/lib.hint	1970-01-01 01:00:00.000000000 +0100
+++ src/icu/source/CYGWIN-PATCHES/lib.hint	2010-09-16 09:34:46.439849900 +0200
@@ -0,0 +1,9 @@
+category: Libs
+requires: libgcc1 libstdc++6
+external-source: icu
+sdesc: "IBM Internationalization Components for Unicode (runtime libs)"
+ldesc: "ICU is a C++ and C library that provides robust and full-featured 
+Unicode and locale support. Only libicu38 is required for runtime.
+The ICU C/C++ project is also called ICU4C when a distinction to 
+the java library ICU4J is necessary.
+For runtime only this package is needed."
