difforig Module-Signature-0.55

2007-09-09  Reini Urban <rurban@x-ray.at>

diff -ub  Module-Signature-0.55/Makefile.PL.orig
--- Module-Signature-0.55/Makefile.PL.orig	2006-07-30 01:11:14.000000000 +0000
+++ Module-Signature-0.55/Makefile.PL	2007-09-09 11:16:51.296875000 +0000
@@ -113,7 +113,7 @@
 
 	return 1 if grep { /^--installdeps/} @ARGV;
 
-	if ( prompt("Import PAUSE and author keys to GnuPG?", 'y' ) =~ /^y/i) {
+	if ( 0 and prompt("Import PAUSE and author keys to GnuPG?", 'y' ) =~ /^y/i) {
 		print 'Importing... ';
 		system 'gpg', '--quiet', '--import', glob('*.pub');
 		print "done.\n";
