difforig perl-current/perl.c

diff -u perl-current/perl.c.orig perl-current/perl.c
--- perl-current/perl.c.orig	2009-03-19 20:23:07.000000000 +0000
+++ perl-current/perl.c	2009-03-24 16:46:58.832875000 +0000
@@ -4921,11 +4921,6 @@
 #  endif
 #endif
 
-#if defined(SITELIB_STEM) && defined(PERL_INC_VERSION_LIST)
-    /* Search for version-specific dirs below here */
-    incpush(SITELIB_STEM, FALSE, TRUE, TRUE, TRUE);
-#endif
-
 #ifdef PERL_VENDORARCH_EXP
     /* vendorarch is always relative to vendorlib on Windows for
      * DLL-based path intuition to work correctly */
@@ -4942,12 +4937,18 @@
 #  endif
 #endif
 
-#ifdef PERL_VENDORLIB_STEM /* Search for version-specific dirs below here */
+#if defined(SITELIB_STEM) && defined(PERL_INC_VERSION_LIST)
+    /* Search for version-specific dirs below here */
+    incpush(SITELIB_STEM, FALSE, TRUE, TRUE, TRUE);
+#endif
+
+#ifdef PERL_VENDORLIB_STEM  && defined(PERL_INC_VERSION_LIST) 
+    /* Search for version-specific dirs below here */
     incpush(PERL_VENDORLIB_STEM, FALSE, TRUE, TRUE, TRUE);
 #endif
 
-#ifdef PERL_OTHERLIBDIRS
-    incpush(PERL_OTHERLIBDIRS, TRUE, TRUE, TRUE, TRUE);
+#ifdef PERL_OTHERLIBDIRS /* CYG PATCH: needs other archlib explicitly */
+    incpush(PERL_OTHERLIBDIRS, FALSE, FALSE, TRUE, FALSE);
 #endif
 
     if (!PL_tainting)
