difforig perl-current/patchlevel.h

diff -u perl-current/patchlevel.h.orig perl-current/patchlevel.h
--- perl-current/patchlevel.h.orig	2008-12-21 12:43:04.000000000 +0000
+++ perl-current/patchlevel.h	2009-03-24 16:48:52.473500000 +0000
@@ -119,6 +119,13 @@
 static const char * const local_patches[] = {
 	NULL
 	,"MAINT35117"
+  	,"CYG11 no-bs"
+  	,"CYG12 no archlib in otherlibdirs"
+ 	,"CYG14 Dynaloader"
+ 	,"CYG15 static-Win32CORE"
+        ,"CYG17 utf8-paths"
+	,"CYG21 LibList-Kid.patch"
+ 	,"Bug#55162 File::Spec::case_tolerant performance"
 	,NULL
 };
 
