difforig perl-5.10.1/patchlevel.h

diff -u perl-5.10.1/patchlevel.h.orig perl-5.10.1/patchlevel.h
--- perl-5.10.1/patchlevel.h.orig	2009-08-18 23:48:11.000000000 +0000
+++ perl-5.10.1/patchlevel.h	2009-12-13 20:45:37.131191700 +0000
@@ -131,6 +131,19 @@
 	NULL
 	PERL_GIT_UNPUSHED_COMMITS    	/* do not remove this line */
         PERL_GIT_UNCOMMITTED_CHANGES	/* do not remove this line */
+ 	,"CYG11 no-bs"
+  	,"CYG12 no archlib in otherlibdirs"
+  	,"CYG14 Dynaloader"
+  	,"CYG15 static-Win32CORE"
+  	,"CYG17 utf8-paths"
+  	,"CYG21 LibList-Kid.patch"
+  	,"CYG22 cygwin-1.7 hints"
+  	,"CYG23 544-stat"
+  	,"CYG24 build man pages"
+  	,"CYG25 rebase_privlib"
+  	,"Module-Build-0.36_13"
+  	,"Bug#55162 CYG18 File::Spec::case_tolerant performance"
+  	,"disable ExtUtils::MakeMaker::Coverage in Sys-Syslog"
 	,NULL
 };
 
