difforig YAML-0.65

2007-09-09  Reini Urban <rurban@x-ray.at>

diff -ub  YAML-0.65/Makefile.PL.orig
--- YAML-0.65/Makefile.PL.orig	2007-04-08 01:33:06.000000000 +0000
+++ YAML-0.65/Makefile.PL	2007-09-09 11:10:51.250000000 +0000
@@ -12,7 +12,8 @@
 _
 
 if (!is_admin()) {
-    exit() unless prompt("Continue installing YAML.pm?", 'y') =~ /^y/i;
+  print "Continuing installing YAML.pm? (prompt skipped)\n";
+  #exit() unless prompt("Continue installing YAML.pm?", 'y') =~ /^y/i;
 }
 
 name            'YAML';
