--- origsrc/postgresql-8.2.11/src/interfaces/libpq/libpq.rc.in	2008-10-31 02:44:09.000000000 +0000
+++ src/postgresql-8.2.11/src/interfaces/libpq/libpq.rc.in	2009-01-07 16:37:51.535125000 +0000
@@ -19,7 +19,7 @@
             VALUE "InternalName", "libpq\0"
             VALUE "LegalCopyright", "Copyright (C) 2005\0"
             VALUE "LegalTrademarks", "\0"
-            VALUE "OriginalFilename", "libpq.dll\0"
+            VALUE "OriginalFilename", "cygpq.dll\0"
             VALUE "ProductName", "PostgreSQL\0"
             VALUE "ProductVersion", "8.2.11\0"
         END
