difforig Crypt-SSLeay-0.58

diff -u  Crypt-SSLeay-0.58/Makefile.PL.orig
--- Crypt-SSLeay-0.58/Makefile.PL.orig	2010-08-24 21:28:46.000000000 +0200
+++ Crypt-SSLeay-0.58/Makefile.PL	2011-02-26 20:23:06.313223000 +0100
@@ -19,6 +19,10 @@
 );
 
 $OPT_DEFAULT ||= $ENV{CRYPT_SSLEAY_DEFAULT};
+if ($^O eq 'cygwin') {
+  $OPT_LIVE_TESTS = 0;
+  $OPT_LIBPATH = '/usr';
+}
 
 # FIND POSSIBLE SSL INSTALLATIONS
 
