commit 31abdf84ad3d78de345ff0851a765076f87b01e5
Author: Reini Urban <rurban@x-ray.at>
Date:   Tue Apr 12 14:27:18 2011 +0200

    APItest coplabel: fix warnings

diff --git a/ext/XS-APItest/APItest.xs b/ext/XS-APItest/APItest.xs
index 4298e7d..4efb22b 100644
--- a/ext/XS-APItest/APItest.xs
+++ b/ext/XS-APItest/APItest.xs
@@ -2333,7 +2333,7 @@ test_coplabel()
     PREINIT:
 	COP *cop;
         char *label;
-        int len, utf8;
+        STRLEN len; U32 utf8;
     CODE:
 	cop = &PL_compiling;
 	Perl_store_cop_label(aTHX_ cop, "foo", 3, 0);
