diff -bu XML-SAX-0.99/Makefile.PL~ XML-SAX-0.99/Makefile.PL
--- XML-SAX-0.99/Makefile.PL~	2011-09-04 16:37:48.000000000 -0500
+++ XML-SAX-0.99/Makefile.PL	2012-07-12 12:42:47.396815600 -0500
@@ -22,7 +22,8 @@
     my $write_ini_ok = 0;
 
     eval { require XML::SAX };
-    if ($@) {
+    my $inc = join(":",@INC);
+    if ($@ or $inc =~ m|usr/src/perl|) {
         $write_ini_ok = 1;
     }
     else {
