diff -urN mhash-0.9.6-orig/CYGWIN-PATCHES/README mhash-0.9.6/CYGWIN-PATCHES/README
--- mhash-0.9.6-orig/CYGWIN-PATCHES/README	2005-01-12 16:54:24.000000000 +0000
+++ mhash-0.9.6/CYGWIN-PATCHES/README	2006-06-07 08:33:05.000000000 +0000
@@ -1,5 +1,5 @@
-mhash-0.9.1-1
------------------------------------------- 
+mhash
+----------------------------------
 MHash - library of hash algorithms
 
 Runtime requirements:
@@ -17,7 +17,7 @@
   fileutils-4.1         or newer
   sed-4.1.2-1           or newer
   m4-1.4-1
-  perl-5                (for automake)
+  perl-5.x              (for automake)
 
 Canonical homepage:
   http://mhash.sourceforge.net/
@@ -30,41 +30,61 @@
 Build instructions:
 
   cd /usr/src
-  unpack mhash-<VERSION>-<CYGREL>-src.tar.bz2
-    if you use setup to install this src package, it will be
-	 unpacked under /usr/src automatically
-  ./mhash-<VERSION>-<CYGREL>.sh prep build install
-
-./mhash-<VERSION>-<CYGREL>.sh all will create:
-  /usr/src/mhash-<VERSION>-<CYGREL>.tar.bz2
-  /usr/src/mhash-<VERSION>-<CYGREL>-src.tar.bz2
+  cygport mhash-<VER>-<REL>.cygport prep compile install check package finish
 
-Or use './mhash-<VERSION>-<CYGREL>.sh prep' to get a patched source directory 
+This will create:
+  /usr/src/mhash-<VER>-<REL>-src.tar.bz2
+  /usr/src/mhash-<VER>-<REL>.tar.bz2
+  /usr/src/mhash-devel-<VER>-<REL>.tar.bz2
+
+Or use 'cygport mhash-<VER>-<REL>.cygport prep' to get a patched source directory
 
 -------------------------------------------------------------------------------
 
 Files included in the binary distribution:
 
   /usr/bin/cygmhash-2.dll
-  /usr/include/mhash.h
+  /usr/include/mutils/mglobal.h
+  /usr/include/mutils/mhash.h
+  /usr/include/mutils/mincludes.h
+  /usr/include/mutils/mtypes.h
+  /usr/include/mutils/mutils.h 
   /usr/lib/libmhash.a
   /usr/lib/libmhash.dll.a
   /usr/lib/libmhash.la
-  /usr/share/doc/Cygwin/mhash-0.9.1.README (this file)
-  /usr/share/doc/mhash-0.9.1/AUTHORS
-  /usr/share/doc/mhash-0.9.1/ChangeLog
-  /usr/share/doc/mhash-0.9.1/COPYING
-  /usr/share/doc/mhash-0.9.1/INSTALL
-  /usr/share/doc/mhash-0.9.1/NEWS
-  /usr/share/doc/mhash-0.9.1/README
-  /usr/share/doc/mhash-0.9.1/THANKS
-  /usr/share/doc/mhash-0.9.1/TODO
+  /usr/share/doc/Cygwin/mhash-0.9.6.README (this file)
+  /usr/share/doc/mhash-0.9.6/AUTHORS
+  /usr/share/doc/mhash-0.9.6/ChangeLog
+  /usr/share/doc/mhash-0.9.6/COPYING
+  /usr/share/doc/mhash-0.9.6/INSTALL
+  /usr/share/doc/mhash-0.9.6/NEWS
+  /usr/share/doc/mhash-0.9.6/README
+  /usr/share/doc/mhash-0.9.6/THANKS
+  /usr/share/doc/mhash-0.9.6/TODO
   /usr/share/man/man3/mhash.3.gz
 
 -------------------------------------------------------------------------------
 
 Port Notes:
 
+----- version mhash-0.9.6-1 -----
+
+changed from gbs to cygport
+fixed lndir build/include/mutils
+
+----- version mhash-0.9.4a-1 ----- (never released)
+
+fixed unpacking into correct mhash-0.9.4a
+removed autom4te.cache
+moved md5-rfc1321.txt, skid2-authentication into 
+  /usr/share/doc/mhash-0.9.4a/
+
+----- version mhash-0.9.2-1 -----
+ 
+All cygwin patches applied upstream. 
+Minor build script improvements.
+Reverted the dll main hack. 
+
 ----- version mhash-0.9.1-1 -----
 
 Maintainer change. Fixed to be mingw independent. All tests PASS.
diff -urN mhash-0.9.6-orig/CYGWIN-PATCHES/mhash-0.9.6-1.cygport mhash-0.9.6/CYGWIN-PATCHES/mhash-0.9.6-1.cygport
--- mhash-0.9.6-orig/CYGWIN-PATCHES/mhash-0.9.6-1.cygport	1970-01-01 00:00:00.000000000 +0000
+++ mhash-0.9.6/CYGWIN-PATCHES/mhash-0.9.6-1.cygport	2006-06-07 09:04:25.320500000 +0000
@@ -0,0 +1,59 @@
+# -*- sh -*-
+DESCRIPTION="library of hash algorithms"
+HOMEPAGE="http://mhash.sourceforge.net/"
+SRC_URI="http://mesh.dl.sourceforge.net/sourceforge/mhash/${PN}-${PV}.tar.bz2"
+#CVS_URI="pserver:anonymous@cvs.mhash.sourceforge.net:/cvsroot/mhash"
+
+MAKEOPTS="${MAKEOPTS} -j1"
+PACKAGE_BUGREPORT="cygwin@cygwin.com"
+CFLAGS="${CFLAGS} -O3"
+LDFLAGS="${LDFLAGS} -Wl,--no-undefined -Wl,--enable-auto-image-base"
+CYGCONF_ARGS=""
+
+PKG_NAMES="${PN} ${PN}-devel"
+PKG_HINTS="${PKG_NAMES}"
+
+# mhash
+PKG_CONTENTS[0]=" \
+usr/bin/cygmhash-2.dll \
+usr/share"
+
+# mhash-devel
+PKG_CONTENTS[1]=" \
+--exclude=usr/bin/cygmhash-2.dll \
+--exclude=usr/share \
+usr"
+
+_lndirs() {
+	FROM=${S}/include/mutils
+	TO=${B}/include/mutils
+	if check_prog lndir
+	then
+		mkdir -p ${TO}
+		lndir -silent ${FROM} ${TO} || error "lndir failed"
+	else
+		cd ${FROM}
+		local d f
+		for d in $(find . -type d)
+		do
+			mkdir -p ${TO}/${d}
+		done
+		for f in $(find . ! -type d)
+		do
+			ln -sf ${FROM}/${f} ${TO}/${f}
+		done
+	fi
+}
+
+src_compile() {
+	cd ${S}
+	cygautoreconf
+	cd ${B}
+
+	# mhash just copies mhash_config.h into include/mutils/
+	_lndirs
+
+	cygconf
+	cygmake
+}
+
diff -urN mhash-0.9.6-orig/CYGWIN-PATCHES/mhash-devel.hint mhash-0.9.6/CYGWIN-PATCHES/mhash-devel.hint
--- mhash-0.9.6-orig/CYGWIN-PATCHES/mhash-devel.hint	1970-01-01 00:00:00.000000000 +0000
+++ mhash-0.9.6/CYGWIN-PATCHES/mhash-devel.hint	2006-06-07 07:36:26.000000000 +0000
@@ -0,0 +1,8 @@
+sdesc: "library of hash algorithms"
+ldesc: "The mhash-devel package contains the header files and libraries needed
+to develop programs that use the mhash library.
+
+Install the mhash-devel package if you want to develop applications that
+will use the libmhash library."
+category: Libs
+requires: cygwin
diff -urN mhash-0.9.6-orig/CYGWIN-PATCHES/mhash.hint mhash-0.9.6/CYGWIN-PATCHES/mhash.hint
--- mhash-0.9.6-orig/CYGWIN-PATCHES/mhash.hint	1970-01-01 00:00:00.000000000 +0000
+++ mhash-0.9.6/CYGWIN-PATCHES/mhash.hint	2006-06-07 07:36:30.000000000 +0000
@@ -0,0 +1,7 @@
+sdesc: "library of hash algorithms (runtime environment)"
+ldesc: "MHash is a library that provides a uniform interface to several hash
+algorithms. It supports the basics for message authentication by following
+rfc2104 (HMAC). It also includes some key generation algorithms which are based
+on hash algorithms. This package contains the runtime environment."
+category: Libs
+requires: cygwin
diff -urN mhash-0.9.6-orig/CYGWIN-PATCHES/setup.hint mhash-0.9.6/CYGWIN-PATCHES/setup.hint
--- mhash-0.9.6-orig/CYGWIN-PATCHES/setup.hint	2005-01-12 16:54:24.000000000 +0000
+++ mhash-0.9.6/CYGWIN-PATCHES/setup.hint	1970-01-01 00:00:00.000000000 +0000
@@ -1,7 +0,0 @@
-sdesc: "MHash - library of hash algorithms (runtime environment)"
-ldesc: "MHash is a library that provides a uniform interface to several hash
-algorithms. It supports the basics for message authentication by following
-rfc2104 (HMAC). It also includes some key generation algorithms which are based
-on hash algorithms. This package contains the runtime environment."
-category: Libs
-requires: cygwin
