diff -urN origsrc/libsigsegv-2.6/CYGWIN-PATCHES/README src/libsigsegv-2.4/CYGWIN-PATCHES/README
--- origsrc/libsigsegv-2.6/CYGWIN-PATCHES/README	1970-01-01 00:00:00.000000000 +0000
+++ src/libsigsegv-2.6/CYGWIN-PATCHES/README	2006-07-29 20:59:51.193225300 +0000
@@ -0,0 +1,58 @@
+libsigsegv
+------------------------------------------
+Handling page faults in user mode
+
+Runtime requirements:
+  cygwin-1.5.11-1 or newer
+
+Build requirements:
+  cygwin-1.5.20-1 or newer
+  gcc-3.3.3-1 or newer
+  binutils-20040725-2 or newer
+
+Canonical homepage:
+  http://libsigsegv.sourceforge.net/
+
+Canonical download:
+  ftp://ftp.gnu.org/pub/gnu/libsigsegv/libsigsegv-2.6.tar.gz
+
+------------------------------------
+
+Build instructions:
+  tar xfj libsigsegv-<VERSION>-<CYGREL>-src.tar.bz2
+    if you use setup to install this src package, it will be
+	 unpacked under /usr/src automatically
+  cd /usr/src
+  cygport libsigsegv-<VERSION>-<CYGREL>.cygport all
+
+This will create:
+  /usr/src/libsigsegv-<VERSION>-<CYGREL>.tar.bz2
+  /usr/src/libsigsegv-<VERSION>-<CYGREL>-src.tar.bz2
+
+Or use 'cygport libsigsegv-<VERSION>-<CYGREL>.cygport prep' 
+to get a patched source directory 
+
+-------------------------------------------
+
+Files included in the binary distribution:
+
+  /usr/include/sigsegv.h
+  /usr/lib/libsigsegv.a
+  /usr/lib/libsigsegv.la
+  /usr/share/doc/Cygwin/libsigsegv-2.6.README
+  /usr/share/doc/libsigsegv-2.6/AUTHORS
+  /usr/share/doc/libsigsegv-2.6/COPYING
+  /usr/share/doc/libsigsegv-2.6/ChangeLog
+  /usr/share/doc/libsigsegv-2.6/INSTALL
+  /usr/share/doc/libsigsegv-2.6/NEWS
+  /usr/share/doc/libsigsegv-2.6/README
+
+------------------
+
+Port Notes:
+
+----------  libsigsegv-2.6-1 -----------
+Initial cygwin release for clisp-2.39-1
+
+Cygwin port maintained by: Reini Urban <rurban@x-ray.at>
+Please address all questions to the Cygwin mailing list at <cygwin@cygwin.com>
diff -urN origsrc/libsigsegv-2.6/CYGWIN-PATCHES/setup.hint src/libsigsegv-2.4/CYGWIN-PATCHES/setup.hint
--- origsrc/libsigsegv-2.6/CYGWIN-PATCHES/setup.hint	1970-01-01 00:00:00.000000000 +0000
+++ src/libsigsegv-2.6/CYGWIN-PATCHES/setup.hint	2006-07-29 20:58:24.677600300 +0000
@@ -0,0 +1,4 @@
+sdesc: "Handling page faults in user mode"
+ldesc: "Small static library for handling page faults in user mode. A page fault occurs when a program tries to access to a region of memory that is currently not available. Mainly required to build clisp from source."
+category: Library
+requires: 
