diff -urN origsrc/ctris-0.42/CYGWIN-PATCHES/README src/ctris-0.42/CYGWIN-PATCHES/README
--- origsrc/ctris-0.42/CYGWIN-PATCHES/README	1970-01-01 00:00:00.000000000 +0000
+++ src/ctris-0.42/CYGWIN-PATCHES/README	2007-07-25 18:56:44.437500000 +0000
@@ -0,0 +1,71 @@
+This had be renamed to ctris.
+
+Runtime requirements:
+  cygwin-1.5.x              or newer
+  libncurses8               or newer  
+  
+Build requirements:
+  cygwin-1.5.x              or newer
+  gcc-3.3.1                 or newer
+  binutils-20030901         or newer
+  make-3.81                 or newer
+  ash-20031007              or newer
+  fileutils-4.1             or newer
+  sed-4.0.7                 or newer
+
+Canonical homepage:
+  http://hackl.dhs.org/ctris/
+
+Canonical download:
+  http://hackl.dhs.org/data/download/download.php?file=ctris-${PV}.tar.bz2
+
+-------------------------------------------------------------------------------
+
+Build instructions:
+  tar xfj ctris-<VERSION>-<CYGREL>-src.tar.bz2
+    if you use setup to install this src package, it will be
+	 unpacked under /usr/src automatically
+  cd /usr/src
+  cygport ctris-<VERSION>-<CYGREL>.cygport all
+
+This will create:
+  /usr/src/ctris-<VERSION>-<CYGREL>.tar.bz2
+  /usr/src/ctris-<VERSION>-<CYGREL>-src.tar.bz2
+
+Or use 'cygport ctris-<VERSION>-<CYGREL>.cygport prep' to get a patched 
+source directory 
+
+-------------------------------------------------------------------------------
+
+Files included in ctris:
+
+  /usr/bin/ctris.exe
+  /usr/share/doc/Cygwin/ctris-0.42.README
+  /usr/share/doc/ctris-0.42/AUTHORS
+  /usr/share/doc/ctris-0.42/COPYING
+  /usr/share/doc/ctris-0.42/INSTALL
+  /usr/share/doc/ctris-0.42/README
+  /usr/share/doc/ctris-0.42/TODO
+  /usr/share/man/man6/ctris.6.gz
+
+-------------------------------------------------------------------------------
+
+Port Notes:
+
+----- version ctris-0.42-1 -----
+
+hacked .cygport
+
+----- version ctris-0.41-1 -----
+
+Renamed from ctetris to ctris
+Converted to cygport
+Fixed make install
+
+----- version ctetris-0.29-1 -----
+
+Initial release
+
+Builds out of the box.
+Cygwin port maintained by: Reini Urban <rurban@x-ray.at>
+Cygwin port questions to the mailinglist <cygwin@cygwin.com>
diff -urN origsrc/ctris-0.42/CYGWIN-PATCHES/setup.hint src/ctris-0.42/CYGWIN-PATCHES/setup.hint
--- origsrc/ctris-0.42/CYGWIN-PATCHES/setup.hint	1970-01-01 00:00:00.000000000 +0000
+++ src/ctris-0.42/CYGWIN-PATCHES/setup.hint	2007-07-25 18:56:03.656250000 +0000
@@ -0,0 +1,6 @@
+sdesc: "Console mode tetris clone"
+ldesc: "ctris is a tetris clone, formerly named ctetris. 
+It works very fast and efficient, so it should be possible to play it even over a 
+slow remote terminal. With highscores. License: GPL"
+category: Games
+requires: libncurses8 cygwin
