--- clamav-0.80/CYGWIN-PATCHES/README.orig	1970-01-01 02:00:00.000000000 +0200
+++ clamav-0.80/CYGWIN-PATCHES/README	2004-09-28 20:48:43.000000000 +0200
@@ -0,1 +1,136 @@
+See /usr/share/doc/clamav-<VER>
+
+The freshclam cronjob and clamd have to be configured manually.
+
+Installation (as service):
+
+/etc/rc.d/init.d/clamd install
+/etc/rc.d/init.d/clamd installcron
+/etc/rc.d/init.d/clamd start
+
+Problems:
+
+* If the service doesn't start, check why the SYSTEM user may not run clamd. 
+  Most likely some file permission problems on the PID, SOCK, LOG or CONF file.
+  Search the cygwin mailinglist for "sysbash" - opening a bash under the SYSTEM 
+  account via AT.
+
+* If the service fails to restart (stop fails), may best suggestion so far 
+  is to reboot. :(
+  Check the cygwin mailing list archives for solutions.
+
+Runtime requirements: (versions given or later)
+  curl-7.11.1-1
+  cygwin-1.5.11-1
+  gmp-4.1.2-1
+  libbz2_1-1.0.2-6
+  minires-0.98-3
+  openssl-0.9.7
+  zlib
+ 
+Build requirements: (versions given or later)
+  cygwin-1.5.11-1
+  gmp-4.1.2-1
+  libbz2_1-1.0.2-5
+  zlib-1.2.1-1
+  openssl-0.9.7
+  minires (?)
+  minires-devel (?)
+  libtool-devel-1.5.10 (if GCC >= 3.3.3)
+  gcc-3.3.3-3
+  binutils-20040725-2
+  make-3.80
+  ash-20031007
+  fileutils-4.1-2
+  sed-4.1.2-1
+
+Canonical homepage:
+  http://clamav.net/
+virusdb update announcements (very frequent!)
+  http://news.gmane.org/gmane.comp.security.virus.clamav.virusdb
+
+Canonical download:
+  http://prdownloads.sourceforge.net/clamav/clamav-<VERSION>.tar.bz2?download
+
+Latest Changes:
+  security fixes, esp. with email handling.
+  /etc/clamav.conf renamed to /etc/clamd.conf
+  See http://cvs.sourceforge.net/viewcvs.py/clamav/clamav-devel/ChangeLog
+
+-------------------------------------------------------------------------------
+
+Build instructions:
+  unpack clamav-<VERSION>.tar.bz2
+    if you use setup to install this src package, it will be
+	 unpacked under /usr/src automatically
+  cd /usr/src
+  ./clamav-<VERSION>-<CYGREL>.sh all
+
+This will create:
+  /usr/src/clamav-<VERSION>-<CYGREL>.tar.bz2
+  /usr/src/clamav-<VERSION>-<CYGREL>-src.tar.bz2
+
+Or use './clamav-<VERSION>-<CYGREL>.sh prep' to get a patched source directory 
+
+-------------------------------------------------------------------------------
+
+Files included in this package:
+
+  /etc/clamd.conf-dist
+  /etc/freshclam.conf-dist
+  /etc/rc.d/init.d/clamd
+  /etc/postinstall/clamav.sh
+  /etc/preremove/clamav.sh
+  /usr/bin/clamav-config
+  /usr/bin/clamdscan.exe
+  /usr/bin/clamscan.exe
+  /usr/bin/freshclam.exe
+  /usr/bin/sigtool.exe
+  /usr/include/clamav.h
+  /usr/lib/libclamav.a
+  /usr/lib/libclamav.la
+  /usr/lib/pkgconfig/libclamav.pc
+  /usr/sbin/clamd.exe
+  /usr/share/clamav/daily.cvd
+  /usr/share/clamav/main.cvd
+  /usr/share/doc/clamav/AUTHORS
+  /usr/share/doc/clamav/ChangeLog
+  /usr/share/doc/clamav/COPYING
+  /usr/share/doc/clamav/FAQ
+  /usr/share/doc/clamav/INSTALL
+  /usr/share/doc/clamav/NEWS
+  /usr/share/doc/clamav/README
+  /usr/share/doc/clamav/TODO
+  /usr/share/doc/clamav/*.pdf
+  /usr/share/doc/clamav/html/*
+  /usr/share/doc/Cygwin/clamav--<VERSION>-<CYGREL>.README
+  /usr/share/man/man1/clamdscan.1.gz
+  /usr/share/man/man1/clamscan.1.gz
+  /usr/share/man/man1/freshclam.1.gz
+  /usr/share/man/man1/sigtool.1.gz
+  /usr/share/man/man5/clamd.conf.5.gz
+  /usr/share/man/man5/freshclam.conf.5.gz
+  /usr/share/man/man8/clamd.8.gz
+
+-------------------------------------------------------------------------------
+
+Port Notes:
+
+----- version clamav-0.80rc3-1 -----
+
+new shared cygclamav-1.dll, thanks to Brian Brubs and Yaakov Selkowitz
+
+----- version clamav-0.80rc2-1 -----
+
+new preremove script to install existing service
+improved /etc/rc.d/init.d/clamd uninstall
+/etc/clamav.conf renamed to /etc/clamd.conf
+
+----- version clamav-0.75.1-3 -----
+
+Initial release
+
+Builds out of the box. No contrib. init.d script added.
+Cygwin port maintained by: Reini Urban <rurban@x-ray.at>
+Cygwin port questions to the mailinglist <cygwin@cygwin.com>
--- clamav-0.80rc3/CYGWIN-PATCHES/clamd.initd.orig	1970-01-01 02:00:00.000000000 +0200
+++ clamav-0.80rc3/CYGWIN-PATCHES/clamd.initd	2004-09-28 15:22:40.000000000 +0200
@@ -0,0 +1,151 @@
+#! /bin/bash
+#
+# crond   Start/Stop the clam antivirus daemon. 0.8x series
+#
+# chkconfig: 2345 90 60
+# description: clamd is a standard UNIX program that scans for Viruses.
+# processname: clamd
+# config: /etc/clamd.conf
+# pidfile: /var/run/clamd.pid
+
+PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
+PREFIX=/usr
+CONFIG=/etc/clamd.conf
+PIDFILE=/var/run/clamd.pid
+LOGFILE=/var/log/clamd.log
+SOCKET=/var/run/clamd.sock
+
+# Source function library.
+if [ -f /etc/init.d/functions ] ; then
+  . /etc/init.d/functions
+fi
+
+RETVAL=0
+
+# See how we were called.
+
+prog="clamd"
+progdir="/usr/sbin"
+DAEMON="$progdir/$prog.exe"
+
+test -f $DAEMON || exit 0
+
+# Source configuration
+if [ -f /etc/sysconfig/$prog ] ; then
+    . /etc/sysconfig/$prog
+fi
+
+start() {
+        echo -n $"Starting $prog: "
+	# check if cygrunsrv process
+        cygrunsrv --start $prog
+        RETVAL=$?
+        echo
+        [ $RETVAL -eq 0 ] && touch $PIDFILE  && echo "done."
+        return $RETVAL
+}
+
+stop() {
+        echo -n $"Stopping $prog: "
+	cygrunsrv --stop $prog
+        if test -r $PIDFILE; then
+            kill `cat $PIDFILE` 2> /dev/null || echo -n " failed"
+        else
+            echo -n " no PID file"
+        fi
+	# this really needs a long time to stop.
+        RETVAL=$?
+        echo "."
+        [ $RETVAL -eq 0 ] && rm -f $PIDFILE $SOCKET && echo "done."
+        return $RETVAL
+}
+
+rhstatus() {
+        # service status
+        cygrunsrv -Q $prog
+        # connect(): No such file or directory
+	/usr/bin/clamdscan /usr/sbin/clamd.exe
+	#status clamd
+}
+
+restart() {
+        echo -n $"Restarting $prog: "
+        $0 stop
+        sleep 1
+        $0 start
+        echo "done."
+}
+
+installcron() {
+    ( crontab -l | grep freshclam >/dev/null 2>&1 ) || ( \
+	echo "add something like this to your cronjob:"
+	echo
+	echo "*  */3 * * *	/usr/bin/freshclam >>/var/log/freshclam.log 2>&1"
+	echo "#*  */3 * * *	/usr/bin/freshclam >>/var/log/freshclam.log 2>&1" >> /var/cron/tabs/$USER
+	kill -s HUP `cat /var/run/cron.pid`
+	crontab -e )
+}
+install() {
+        freshclam
+        echo -n $"Installing $prog daemon: "
+	# some safety measures
+	touch $LOGFILE $PIDFILE
+	chgrp 18 $LOGFILE $PIDFILE
+	chmod g+w $LOGFILE $PIDFILE
+	rm -f $SOCKET
+	#it was compiled with uid=18
+	cygrunsrv --install $prog --path $DAEMON -e CYGWIN="$CYGWIN" --disp "CYGWIN `$DAEMON --version`"
+        echo "done."
+}
+uninstall() {
+        echo -n $"Uninstalling $prog daemon: "
+	stop
+	cygrunsrv --remove $prog
+        echo "done."
+}
+reload() {
+        echo -n $"Reloading $prog daemon configuration: "
+	echo "unsupported. $0 restart"
+	return 1
+
+	/usr/bin/kill -HUP `cat $PIDFILE`
+        RETVAL=$?
+        [ $RETVAL -eq 0 ] && echo "done."
+        RETVAL=$?
+        return $RETVAL
+}
+
+case "$1" in
+  start)
+        start
+        ;;
+  stop)
+        stop
+        ;;
+  restart)
+        restart
+        ;;
+  reload)
+        reload
+        ;;
+  install)
+        install
+        ;;
+  installcron)
+        installcron
+        ;;
+  uninstall)
+        uninstall
+        ;;
+  status)
+        rhstatus
+        ;;
+  condrestart)
+        [ -f $PIDFILE ] && restart || :
+        ;;
+  *)
+        echo $"Usage: $0 {start|stop|status|install|uninstall|installcron|restart|condrestart}"
+        exit 1
+esac
+
+exit $?
--- clamav-0.80rc3/CYGWIN-PATCHES/postinstall.sh.orig	1970-01-01 02:00:00.000000000 +0200
+++ clamav-0.80rc3/CYGWIN-PATCHES/postinstall.sh	2004-09-28 15:22:40.000000000 +0200
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+for f in clamd freshclam; do
+  if [ ! -e /etc/$f.conf ]; then mv /etc/$f.conf-dist /etc/$f.conf; fi
+done
+
+# update the database (/usr/share/clamav/)
+/usr/bin/freshclam
--- clamav-0.80rc3/CYGWIN-PATCHES/preremove.sh.orig	1970-01-01 02:00:00.000000000 +0200
+++ clamav-0.80rc3/CYGWIN-PATCHES/preremove.sh	2004-09-28 15:22:40.000000000 +0200
@@ -0,0 +1,4 @@
+#!/bin/sh
+initd=/etc/rc.d/init.d/clamd
+# uninstall the service
+if [ -e $initd ]; then $initd uninstall; fi
--- clamav-0.80rc3/etc/clamd.conf.orig	2004-09-27 23:05:43.000000000 +0200
+++ clamav-0.80rc3/etc/clamd.conf	2004-09-28 22:43:18.168682000 +0200
@@ -5,13 +5,14 @@
 
 
 # Comment or remove the line below.
-Example
+#Example
 
 # Uncomment this option to enable logging.
 # LogFile must be writable for the user running daemon.
 # A full path is required.
 # Default: disabled
 #LogFile /tmp/clamd.log
+LogFile /var/log/clamd.log
 
 # By default the log file is locked for writing - the lock protects against
 # running clamd multiple times (if want to run another clamd, please
@@ -54,11 +55,11 @@
 # This option allows you to save a process identifier of the listening
 # daemon (main thread).
 # Default: disabled
-#PidFile /var/run/clamd.pid
+PidFile /var/run/clamd.pid
 
 # Optional path to the global temporary directory.
 # Default: system specific (usually /tmp or /var/tmp).
-#TemporaryDirectory /var/tmp
+TemporaryDirectory /tmp
 
 # Path to the database directory.
 # Default: hardcoded (depends on installation options)
@@ -69,7 +70,7 @@
 
 # Path to a local socket file the daemon will listen on.
 # Default: disabled
-LocalSocket /tmp/clamd
+LocalSocket /var/run/clamd.sock
 
 # Remove stale socket after unclean shutdown.
 # Default: disabled
--- clamav-0.80rc3/libclamav/main.c.orig	1970-01-01 02:00:00.000000000 +0200
+++ clamav-0.80rc3/libclamav/main.c	2004-09-28 21:05:44.329549600 +0200
@@ -0,0 +1,2 @@
+void main() {}
+
--- clamav-0.80rc3/libclamav/Makefile.am.orig	2004-09-07 23:09:56.000000000 +0200
+++ clamav-0.80rc3/libclamav/Makefile.am	2004-09-28 23:01:28.826596400 +0200
@@ -20,7 +20,7 @@
 
 libclamav_la_LIBADD = @LIBCLAMAV_LIBS@
 
-libclamav_la_LDFLAGS = @TH_SAFE@ -version-info @LIBCLAMAV_VERSION@
+libclamav_la_LDFLAGS = @TH_SAFE@ -version-info @LIBCLAMAV_VERSION@ -no-undefined
 
 include_HEADERS = clamav.h
 
@@ -116,6 +116,7 @@
 	line.c \
 	line.h \
 	untar.c \
-	untar.h
+	untar.h \
+	main.c
 
 lib_LTLIBRARIES = libclamav.la
