diff -urN origsrc/fcgi-2.4.0/CYGWIN-PATCHES/README src/fcgi-2.4.0/CYGWIN-PATCHES/README
--- origsrc/fcgi-2.4.0/CYGWIN-PATCHES/README	1970-01-01 00:00:00.000000000 +0000
+++ src/fcgi-2.4.0/CYGWIN-PATCHES/README	2006-08-05 11:11:29.808250000 +0000
@@ -0,0 +1,132 @@
+fcgi
+------------------------------------------
+FastCGI A High-Performance Gateway Interface.
+
+Static library and a cgi-wrapper to create fastcgi enabled applications. 
+End users will most likely not need that.
+FastCGI is a fast, open, and secure Web server interface that
+solves the performance problems inherent in CGI without introducing
+any of the new problems associated with writing applications to
+lower-level Web server APIs. Modules to support FastCGI can be plugged
+into Web server APIs such as Apache API, NSAPI, and ISAPI."
+
+
+This library builds only as static library so far.
+
+Runtime requirements:
+  cygwin-1.5.20-1 or newer
+
+Build requirements:
+  cygwin
+  gcc-core
+  automake
+  autoconf
+  binutils
+
+Canonical homepage:
+  http://fastcgi.com/
+
+Canonical download:
+  http://fastcgi.com/dist/fcgi-2.4.0.tar.gz
+
+------------------------------------
+
+Build instructions:
+  tar xfj fcgi-<VERSION>-<CYGREL>-src.tar.bz2
+    if you use setup to install this src package, it will be
+	 unpacked under /usr/src automatically
+  cd /usr/src
+  cygport fcgi-<VERSION>-<CYGREL>.cygport all
+
+This will create:
+  /usr/src/fcgi-<VERSION>-<CYGREL>.tar.bz2
+  /usr/src/fcgi-<VERSION>-<CYGREL>-src.tar.bz2
+
+Or use 'cygport fcgi-<VERSION>-<CYGREL>.cygport prep' 
+to get a patched source directory 
+
+-------------------------------------------
+
+Files included in the binary distribution:
+
+  /usr/bin/cgi-fcgi.exe
+  /usr/include/fastcgi.h
+  /usr/include/fcgi_config.h
+  /usr/include/fcgi_stdio.h
+  /usr/include/fcgiapp.h
+  /usr/include/fcgimisc.h
+  /usr/include/fcgio.h
+  /usr/include/fcgios.h
+  /usr/lib/fcgi/fcgi-bin/authorizer.exe
+  /usr/lib/fcgi/fcgi-bin/echo-cpp.exe
+  /usr/lib/fcgi/fcgi-bin/echo-x.exe
+  /usr/lib/fcgi/fcgi-bin/echo.exe
+  /usr/lib/fcgi/fcgi-bin/log-dump.exe
+  /usr/lib/fcgi/fcgi-bin/size.exe
+  /usr/lib/fcgi/fcgi-bin/threaded.exe
+  /usr/lib/libfcgi++.a
+  /usr/lib/libfcgi++.la
+  /usr/lib/libfcgi.a
+  /usr/lib/libfcgi.la
+  /usr/share/doc/Cygwin/fcgi-2.4.0.README
+  /usr/share/doc/fcgi-2.4.0/INSTALL
+  /usr/share/doc/fcgi-2.4.0/LICENSE.TERMS
+  /usr/share/doc/fcgi-2.4.0/README
+  /usr/share/doc/fcgi-2.4.0/examples/authorizer.c
+  /usr/share/doc/fcgi-2.4.0/examples/authorizer.mak
+  /usr/share/doc/fcgi-2.4.0/examples/echo-cpp.cpp
+  /usr/share/doc/fcgi-2.4.0/examples/echo-cpp.mak
+  /usr/share/doc/fcgi-2.4.0/examples/echo-x.c
+  /usr/share/doc/fcgi-2.4.0/examples/echo.c
+  /usr/share/doc/fcgi-2.4.0/examples/echo.mak
+  /usr/share/doc/fcgi-2.4.0/examples/echox.mak
+  /usr/share/doc/fcgi-2.4.0/examples/log-dump.c
+  /usr/share/doc/fcgi-2.4.0/examples/size.c
+  /usr/share/doc/fcgi-2.4.0/examples/size.mak
+  /usr/share/doc/fcgi-2.4.0/examples/threaded.c
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-prog-guide/ap_guida.htm
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-prog-guide/ap_guide.htm
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-prog-guide/apaman.htm
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-prog-guide/ch1inta1.gif
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-prog-guide/ch1intra.gif
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-prog-guide/ch1intro.htm
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-prog-guide/ch2c.htm
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-prog-guide/ch3perl.htm
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-prog-guide/ch4tcl.htm
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-prog-guide/cover.htm
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-prog-guide/covera.gif
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-whitepaper/fastcgi.htm
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-whitepaper/img00001.gif
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-whitepaper/img00002.gif
+  /usr/share/doc/fcgi-2.4.0/html/fastcgi-whitepaper/img00003.gif
+  /usr/share/doc/fcgi-2.4.0/html/fcgi-devel-kit.htm
+  /usr/share/doc/fcgi-2.4.0/html/fcgi-java.htm
+  /usr/share/doc/fcgi-2.4.0/html/fcgi-perf.htm
+  /usr/share/doc/fcgi-2.4.0/html/fcgi-perl.htm
+  /usr/share/doc/fcgi-2.4.0/html/fcgi-spec.html
+  /usr/share/doc/fcgi-2.4.0/html/fcgi-tcl.htm
+  /usr/share/doc/fcgi-2.4.0/html/omi-logo.gif
+  /usr/share/doc/fcgi-2.4.0/html/overview.html
+  /usr/share/doc/fcgi-2.4.0/html/www5-api-workshop.html
+  /usr/share/doc/fcgi-2.4.0/images/aplib-hd.gif
+  /usr/share/doc/fcgi-2.4.0/images/divider.gif
+  /usr/share/doc/fcgi-2.4.0/images/fcgi-hd.gif
+  /usr/share/doc/fcgi-2.4.0/images/mail-hd.gif
+  /usr/share/doc/fcgi-2.4.0/images/navbar.gif
+  /usr/share/doc/fcgi-2.4.0/images/serv-hd.gif
+  /usr/share/doc/fcgi-2.4.0/images/words-hd.gif
+  /usr/share/man/man1/cgi-fcgi.1.gz
+  /usr/share/man/man3/FCGI_Accept.3.gz
+  /usr/share/man/man3/FCGI_Finish.3.gz
+  /usr/share/man/man3/FCGI_SetExitStatus.3.gz
+  /usr/share/man/man3/FCGI_StartFilterData.3.gz
+
+------------------
+
+Port Notes:
+
+----------  fcgi-2.4.0-1 -----------
+Initial cygwin release. (for clisp-2.39-1)
+
+Cygwin port maintained by: Reini Urban <rurban@x-ray.at>
+Please address all questions to the Cygwin mailing list at <cygwin@cygwin.com>
diff -urN origsrc/fcgi-2.4.0/CYGWIN-PATCHES/setup.hint src/fcgi-2.4.0/CYGWIN-PATCHES/setup.hint
--- origsrc/fcgi-2.4.0/CYGWIN-PATCHES/setup.hint	1970-01-01 00:00:00.000000000 +0000
+++ src/fcgi-2.4.0/CYGWIN-PATCHES/setup.hint	2006-08-05 11:14:28.308250000 +0000
@@ -0,0 +1,10 @@
+sdesc: "FastCGI A High-Performance Gateway Interface"
+ldesc: "Static library to create fastcgi enabled applications. 
+End users will most likely not need that.
+FastCGI is a fast, open, and secure Web server interface that
+solves the performance problems inherent in CGI without introducing
+any of the new problems associated with writing applications to
+lower-level Web server APIs. Modules to support FastCGI can be plugged
+into Web server APIs such as Apache API, NSAPI, and ISAPI."
+category: Devel Web
+requires: cygwin
