diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/mathomatic-12.8.5/README src/mathomatic-12.8.5/README
--- origsrc/mathomatic-12.8.5/README	1970-01-01 01:00:00.000000000 +0100
+++ src/mathomatic-12.8.5/README	2008-01-21 10:04:41.765625000 +0100
@@ -0,0 +1,160 @@
+mathomatic
+-------------------------------------------------------------------------------
+Algebraic console.
+
+Runtime requirements:
+  cygwin-1.5.x              or newer
+  libreadline6
+  libncurses8
+  
+Build requirements:
+  cygwin
+  gcc
+  binutils
+  make
+  bash
+  fileutils
+  sed
+  libncurses
+  libreadline
+
+Canonical homepage:
+  http://www.mathomatic.org
+
+Canonical download:
+  http://mathomatic.orgserve.de/am.tar.bz2 or 
+  http://www.panix.com/~gesslein/mathomatic-<VERSION>.tar.bz2
+
+-------------------------------------------------------------------------------
+
+Build instructions:
+  tar xfj mathomatic-<VERSION>-<CYGREL>-src.tar.bz2
+    if you use setup to install this src package, it will be
+    unpacked under /usr/src automatically
+  cd /usr/src
+  cygport mathomatic-<VERSION>-<CYGREL>.cygport all
+
+This will create:
+  /usr/src/mathomatic-<VERSION>-<CYGREL>.tar.bz2
+  /usr/src/mathomatic-<VERSION>-<CYGREL>-src.tar.bz2
+
+Or use 'cygport mathomatic-<VERSION>-<CYGREL>.cygport prep' to 
+get a patched source directory.
+
+-------------------------------------------------------------------------------
+
+Files included in mathomatic:
+
+  /usr/bin/mathomatic.exe
+  /usr/bin/mathomatic_secure.exe
+  /usr/include/mathomatic.h
+  /usr/lib/libmathomatic.a
+  /usr/share/doc/Cygwin/mathomatic-<VERSION>.README (this file)
+  /usr/share/doc/mathomatic-<VERSION>/COPYING
+  /usr/share/doc/mathomatic-<VERSION>/README.txt
+  /usr/share/doc/mathomatic-<VERSION>/changes.txt
+  /usr/share/doc/mathomatic-<VERSION>/mathomatic.ico (desktop icon)
+  /usr/share/doc/mathomatic-<VERSION>/html/am.html
+  /usr/share/doc/mathomatic-<VERSION>/html/favicon.ico
+  /usr/share/doc/mathomatic-<VERSION>/html/greenfade.png
+  /usr/share/doc/mathomatic-<VERSION>/html/index.html
+  /usr/share/doc/mathomatic-<VERSION>/html/manpage.html
+  /usr/share/doc/mathomatic-<VERSION>/html/manual.html
+  /usr/share/doc/mathomatic-<VERSION>/html/notes.html
+  /usr/share/doc/mathomatic-<VERSION>/html/open_book_nae_02.png
+  /usr/share/doc/mathomatic-<VERSION>/tasks.txt
+  /usr/share/man/man1/mathomatic.1.gz
+
+-------------------------------------------------------------------------------
+
+Port Notes:
+
+----- version mathomatic-12.8.5-1 -----
+added library
+----- version mathomatic-12.8.0-1 -----
+none
+----- version mathomatic-12.7.3-1 -----
+none
+----- version mathomatic-12.7.0-1 -----
+none
+----- version mathomatic-12.6.10-1 -----
+none
+----- version mathomatic-12.6.7-1 -----
+none
+----- version mathomatic-12.6.6-1 -----
+none
+----- version mathomatic-12.6.4-1 -----
+none
+----- version mathomatic-12.6.3-1 -----
+none
+----- version mathomatic-12.6.1-1 -----
+moved html_docs into html
+
+----- version mathomatic-12.6.0-1 -----
+none
+
+----- version mathomatic-12.5.22-1 -----
+none
+
+----- version mathomatic-12.5.21-1 (never uploaded) -----
+none
+
+----- version mathomatic-12.5.20-1 -----
+
+added *.ico to /usr/share/doc/mathomatic-<VERSION>/
+
+----- version mathomatic-12.5.17-1 -----
+
+none
+
+----- version mathomatic-12.5.16-1 ----- (not released)
+
+changed from gbs to cygport
+added /usr/bin/mathomatic_secure.exe 
+
+----- version mathomatic-12.5.14-1 -----
+
+none
+
+----- version mathomatic-12.5.12-1 -----
+
+none
+
+----- version mathomatic-12.5.11-1 -----
+
+improved makefile.cygwin
+
+----- version mathomatic-12.5.10-1 -----
+
+changed am.tgz to am.tar.bz2, opt_decomp=j
+
+----- version mathomatic-12.5.2-1 -----
+
+updated Makefile.cygwin and README
+
+----- version mathomatic-12.1e-1 -----
+
+none
+
+----- version mathomatic-12.1-1 -----
+
+none
+
+----- version mathomatic-11.5-1 -----
+
+patched \r\n => \n, added -DUNIX for termios handlers, added -O2
+
+----- version mathomatic-11.3f-2 -----
+ 
+Maintainer fixed his basedir. readline and curses added.
+
+----- version mathomatic-11.3d-1 -----
+
+Initial release (not packaged)
+
+This is just the mathomatic client from the mathomatic package. 
+Not the samples for prime number generation: twin, sumsq, primes and around. 
+If you want these also, get the src packages and compile them by yourself.
+
+Cygwin port maintained by: Reini Urban <rurban@x-ray.at>
+Cygwin port questions to the mailinglist <cygwin@cygwin.com>
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/mathomatic-12.8.5/list.c src/mathomatic-12.8.5/list.c
--- origsrc/mathomatic-12.8.5/list.c	2008-01-01 14:51:23.000000000 +0100
+++ src/mathomatic-12.8.5/list.c	2008-01-21 10:04:41.984375000 +0100
@@ -124,7 +124,7 @@
 		len += fprintf(gfp, ";");
 	}
 	if (high_prec) {
-#if	CYGWIN
+#if	WIN32
 		fprintf(gfp, "\r\n");
 #else
 		fprintf(gfp, "\n");
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/mathomatic-12.8.5/makefile.cygwin src/mathomatic-12.8.5/makefile.cygwin
--- origsrc/mathomatic-12.8.5/makefile.cygwin	2007-11-19 13:36:04.000000000 +0100
+++ src/mathomatic-12.8.5/makefile.cygwin	2008-01-21 10:09:58.281250000 +0100
@@ -1,20 +1,34 @@
 # Makefile for compiling and testing Mathomatic with Cygwin for Windows.
 
 VERSION		= `cat VERSION`
-CFLAGS		= -O -DCYGWIN -DVERSION=\"$(VERSION)\"
-LDFLAGS		= 
+CFLAGS		= -Wuninitialized -Wshadow -Wformat -Wparentheses -Wcast-align # gcc specific flags
+CFLAGS		+= -O2 -DCYGWIN -DUNIX -DVERSION=\"$(VERSION)\" # general cc flags
+LDFLAGS		+= 
 LIBS		= -lm
 
 CFLAGS		+= -DREADLINE
 LIBS		+= -lreadline -lcurses
 
-AOUT		= mathomatic.exe
+DESTDIR		=
+prefix		= /usr/local
+bindir		= $(prefix)/bin
+mandir		= $(prefix)/share/man
+docdir		= $(prefix)/share/doc
+
+DOC		= doc/mathomatic.1.html doc/matho-primes.1.html doc/matho-pascal.1.html doc/matho-sumsq.1.html
+AOUT    	= mathomatic.exe
+BINS		= mathomatic.exe mathomatic_secure.exe
 OBJECTS		= main.o globals.o am.o solve.o help.o parse.o cmds.o simplify.o factor.o super.o \
 		  unfactor.o poly.o diff.o integrate.o complex.o complex_lib.o list.o gcd.o factor_int.o
 
-all: $(AOUT)
+MAN1		= mathomatic.1
+DOCS		= VERSION AUTHORS COPYING README.txt changes.txt mathomatic.ico
+
+all: $(BINS)
 	@echo Make completed.
 
+doc: $(DOC)
+
 check test:
 	cd tests && time ../$(AOUT) -t all 0<&- >test.out && fc all.out test.out
 	@echo All tests passed.
@@ -25,7 +39,25 @@
 $(OBJECTS): includes.h am.h externs.h complex.h proto.h VERSION
 
 $(AOUT): $(OBJECTS)
-	$(CC) $(LDFLAGS) $(OBJECTS) $(LIBS) -o $(AOUT)
+	$(CC) $(LDFLAGS) $(OBJECTS) -o $@ $(LIBS)
+
+mathomatic_secure.exe: $(OBJECTS)
+	make -f makefile.cygwin clean
+	$(CC) -c -DSECURE -DTIMEOUT_SECONDS=3600 $(CFLAGS) *.c
+	$(CC) $(LDFLAGS) $(OBJECTS) -o $@ $(LIBS)
+	make -f makefile.cygwin clean
+
+install: $(AOUT)
+	install -d $(DESTDIR)$(prefix)/bin/
+	install -d $(DESTDIR)$(mandir)/man1
+	install -d $(DESTDIR)$(docdir)/mathomatic-$(VERSION)
+	install -d $(DESTDIR)$(docdir)/mathomatic-$(VERSION)/html
+	install -m 0755 $(BINS) $(DESTDIR)$(prefix)/bin/
+	install -m 0644 mathomatic.1 $(DESTDIR)$(mandir)/man1
+	install -m 0644 mathomatic.ico $(DESTDIR)$(docdir)/mathomatic-$(VERSION)
+	install -m 0644 *.txt $(DESTDIR)$(docdir)/mathomatic-$(VERSION)
+	install -m 0644 doc/* $(DESTDIR)$(docdir)/mathomatic-$(VERSION)/html
+	@echo Install completed.
 
 clean:
 	rm -f *.o
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/mathomatic-12.8.5/makefile.lib src/mathomatic-12.8.5/makefile.lib
--- origsrc/mathomatic-12.8.5/makefile.lib	2007-11-25 21:18:37.000000000 +0100
+++ src/mathomatic-12.8.5/makefile.lib	2008-01-21 10:12:02.921875000 +0100
@@ -4,10 +4,11 @@
 
 VERSION		= `cat VERSION`
 CFLAGS		+= -Wuninitialized -Wshadow -Wformat -Wparentheses -Wcast-align # gcc specific flags
-CFLAGS		+= -O -DLIBRARY -DUNIX -DVERSION=\"$(VERSION)\" # general cc flags
+CFLAGS		+= -O2 -DLIBRARY -DCYGWIN -DUNIX -DVERSION=\"$(VERSION)\" # general cc flags
 LDFLAGS		+= 
 LIBS		+= -lm
 
+DESTDIR		=
 prefix		?= /usr/local
 
 AOUT		= mathomatic # The name of the library test executable to create.
@@ -30,8 +31,10 @@
 	$(CC) $(LDFLAGS) lib/test.o $(LIB) $(LIBS) -o $(AOUT)
 
 install:
-	install -m 0644 $(LIB) $(prefix)/lib
-	install -m 0644 lib/mathomatic.h $(prefix)/include
+	install -d $(DESTDIR)$(prefix)/lib
+	install -d $(DESTDIR)$(prefix)/include
+	install -m 0644 $(LIB) $(DESTDIR)$(prefix)/lib/
+	install -m 0644 lib/mathomatic.h $(DESTDIR)$(prefix)/include/
 
 clean:
 	rm -f *.o
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x Makefile.in.in -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x mdate-sh -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x texinfo.tex -x ylwrap -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x omf.make -x xmldocs.make origsrc/mathomatic-12.8.5/setup.hint src/mathomatic-12.8.5/setup.hint
--- origsrc/mathomatic-12.8.5/setup.hint	1970-01-01 01:00:00.000000000 +0100
+++ src/mathomatic-12.8.5/setup.hint	2008-01-21 10:04:41.765625000 +0100
@@ -0,0 +1,9 @@
+sdesc: "Computer Algebra System"
+ldesc: "Mathomatic is a highly portable,general purpose symbolic math program
+that can solve, simplify, combine, differentiate, integrate, and
+compare algebraic equations. It can do standard, complex number, and
+polynomial arithmetic. It is designed to be as general as possible,
+with few options. It is a console mode application that compiles and
+installs easily. License: LPGL 2.1"
+category: Math
+requires: libreadline6 libncurses8 cygwin
