diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x 'mathomatic-12.8.5-1.cygport~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x config.rpath -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/mathomatic-12.9.2/am.h src/mathomatic-12.9.2/am.h
--- origsrc/mathomatic-12.9.2/am.h	2008-03-13 16:24:40.000000000 +0100
+++ src/mathomatic-12.9.2/am.h	2008-04-17 16:20:20.000000000 +0200
@@ -50,7 +50,7 @@
 #define	MAX_CMD_LEN	PATH_MAX			/* maximum command line length (not equations), also max file name length */
 #define	MAX_PROMPT_LEN	STANDARD_SCREEN_COLUMNS		/* maximum length of prompts */
 
-#if	CYGWIN
+#if	WIN32
 #define RC_FILE		"mathomatic.rc"	/* file of set options in the Mathomatic directory under Windows */
 #else
 #define RC_FILE		".mathomaticrc"	/* file of set options read at startup from $HOME */
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x 'mathomatic-12.8.5-1.cygport~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x config.rpath -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/mathomatic-12.9.2/list.c src/mathomatic-12.9.2/list.c
--- origsrc/mathomatic-12.9.2/list.c	2008-03-16 13:08:13.000000000 +0100
+++ src/mathomatic-12.9.2/list.c	2008-04-17 16:10:01.000000000 +0200
@@ -124,7 +124,7 @@
 		len += fprintf(gfp, ";");
 	}
 	if (high_prec || export_flag) {
-#if	CYGWIN
+#if	WIN32
 		fprintf(gfp, "\r\n");	/* might be redirecting to a Microsoft text file */
 #else
 		fprintf(gfp, "\n");
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x 'mathomatic-12.8.5-1.cygport~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x config.rpath -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/mathomatic-12.9.2/main.c src/mathomatic-12.9.2/main.c
--- origsrc/mathomatic-12.9.2/main.c	2008-03-22 01:14:12.000000000 +0100
+++ src/mathomatic-12.9.2/main.c	2008-04-17 16:22:56.062500000 +0200
@@ -262,8 +262,8 @@
 	char	*cp;
 	int	rv = true;
 
-#if	CYGWIN
-	snprintf(buf, sizeof(buf), "%s/%s", dir_path, RC_FILE);
+#if	WIN32
+	snprintf(buf, sizeof(buf), "%s\\%s", dir_path, RC_FILE);
 	fp = fopen(buf, "r");
 	if (fp == NULL) {
 		return true;
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x 'mathomatic-12.8.5-1.cygport~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x config.rpath -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/mathomatic-12.9.2/makefile.cygwin src/mathomatic-12.9.2/makefile.cygwin
--- origsrc/mathomatic-12.9.2/makefile.cygwin	2008-04-08 00:26:04.000000000 +0200
+++ src/mathomatic-12.9.2/makefile.cygwin	2008-04-17 16:23:19.906250000 +0200
@@ -1,20 +1,34 @@
 # Makefile for compiling and testing Mathomatic with Cygwin for MS-Windows.
 
 VERSION		= `cat VERSION`
-CFLAGS		= -O -DCYGWIN -DVERSION=\"$(VERSION)\"
-LDFLAGS		= 
+CFLAGS		= -Wuninitialized -Wshadow -Wformat -Wparentheses -Wcast-align # gcc specific flags
+CFLAGS		+= -O2 -DCYGWIN -DUNIX -DVERSION=\"$(VERSION)\" # general cc flags
+LDFLAGS		+= 
 LIBS		= -lm
 
 CFLAGS		+= -DREADLINE
 LIBS		+= -lreadline -lcurses
 
-AOUT		= mathomatic.exe
+DESTDIR		=
+prefix		?= /usr/local
+bindir		= $(prefix)/bin
+mandir		= $(prefix)/share/man
+docdir		= $(prefix)/share/doc
+
+DOC		= doc/mathomatic.1.html doc/matho-primes.1.html doc/matho-pascal.1.html doc/matho-sumsq.1.html
+AOUT    	= mathomatic.exe
+BINS		= mathomatic.exe mathomatic_secure.exe
 OBJECTS		= main.o globals.o am.o solve.o help.o parse.o cmds.o simplify.o factor.o super.o \
 		  unfactor.o poly.o diff.o integrate.o complex.o complex_lib.o list.o gcd.o factor_int.o
 
-all: $(AOUT)
+MAN1		= mathomatic.1
+DOCS		= VERSION AUTHORS COPYING README.txt changes.txt mathomatic.ico
+
+all: $(BINS)
 	@echo Make completed.
 
+doc: $(DOC)
+
 check test:
 	cd tests && time ../$(AOUT) -t all 0<&- >test.out && fc all.out test.out
 	@echo All tests passed.
@@ -25,7 +39,27 @@
 $(OBJECTS): includes.h am.h externs.h complex.h proto.h VERSION
 
 $(AOUT): $(OBJECTS)
-	$(CC) $(LDFLAGS) $(OBJECTS) $(LIBS) -o $(AOUT)
+	$(CC) $(LDFLAGS) $(OBJECTS) -o $@ $(LIBS)
+
+mathomatic_secure.exe: $(OBJECTS)
+	make -f makefile.cygwin clean
+	$(CC) -c -DSECURE -DTIMEOUT_SECONDS=3600 $(CFLAGS) *.c
+	$(CC) $(LDFLAGS) $(OBJECTS) -o $@ $(LIBS)
+	make -f makefile.cygwin clean
+
+install: $(AOUT)
+	install -d $(DESTDIR)$(prefix)/bin
+	install -d $(DESTDIR)$(prefix)/etc
+	install -d $(DESTDIR)$(mandir)/man1
+	install -d $(DESTDIR)$(docdir)/mathomatic-$(VERSION)
+	install -d $(DESTDIR)$(docdir)/mathomatic-$(VERSION)/html
+	install -m 0755 $(BINS) $(DESTDIR)$(prefix)/bin/
+	install -m 0644 mathomatic.rc $(DESTDIR)/home/$USER/.mathomaticrc
+	install -m 0644 mathomatic.1 $(DESTDIR)$(mandir)/man1
+	install -m 0644 mathomatic.ico $(DESTDIR)$(docdir)/mathomatic-$(VERSION)
+	install -m 0644 *.txt $(DESTDIR)$(docdir)/mathomatic-$(VERSION)
+	install -m 0644 doc/* $(DESTDIR)$(docdir)/mathomatic-$(VERSION)/html
+	@echo Install completed.
 
 clean:
 	rm -f *.o
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x 'mathomatic-12.8.5-1.cygport~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x config.rpath -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/mathomatic-12.9.2/makefile.lib src/mathomatic-12.9.2/makefile.lib
--- origsrc/mathomatic-12.9.2/makefile.lib	2008-03-21 11:10:03.000000000 +0100
+++ src/mathomatic-12.9.2/makefile.lib	2008-04-17 16:16:39.437500000 +0200
@@ -2,10 +2,11 @@
 # Be sure to "make clean" between using different makefiles.
 
 VERSION		= `cat VERSION`
-CFLAGS		+= -O -DLIBRARY -DUNIX -DVERSION=\"$(VERSION)\" # C compiler flags
+CFLAGS		+= -O2 -DLIBRARY -DCYGWIN -DUNIX -DVERSION=\"$(VERSION)\" # general cc flags
 LDFLAGS		+= # linker flags
 LIBS		+= -lm # libraries to link
 
+DESTDIR		=
 prefix		?= /usr/local
 
 AOUT		= mathomatic # The name of the library test executable file to create.
@@ -29,8 +30,10 @@
 	$(CC) $(LDFLAGS) lib/test.o $(LIB) $(LIBS) -o $(AOUT)
 
 install:
-	install -m 0644 $(LIB) $(prefix)/lib
-	install -m 0644 $(HEADERS) $(prefix)/include
+	install -d $(DESTDIR)$(prefix)/lib
+	install -d $(DESTDIR)$(prefix)/include
+	install -m 0644 $(LIB) $(DESTDIR)$(prefix)/lib/
+	install -m 0644 lib/mathomatic.h $(DESTDIR)$(prefix)/include/
 
 clean:
 	rm -f *.o
