Index: parrot-svn/lib/Parrot/Configure/Compiler.pm
===================================================================
--- parrot-svn.orig/lib/Parrot/Configure/Compiler.pm
+++ parrot-svn/lib/Parrot/Configure/Compiler.pm
@@ -191,15 +191,38 @@ defaults to true.
 
 =item conditioned_lines
 
-If C<conditioned_lines> is true, then lines in the file that begin with:
-C<#CONDITIONED_LINE(var):> are skipped if the C<var> condition is false. Lines
-that begin with C<#INVERSE_CONDITIONED_LINE(var):> are skipped if
-the C<var> condition is true.  For instance:
+If conditioned_lines is true, then lines in the file that begin with
+C<#+(var):> are skipped if the var condition is false.
+Lines that begin with C<#-(var):> are skipped if the var condition is true.
+For legacy the old syntax #CONDITIONED_LINE(var): and #INVERSE_CONDITIONED_LINE(var):
+is also supported.
+
+A condition var may be a single keyword, which is true if a config key is true
+or equal to the platform name,
+or a logical combination or (and var1 var2...) or (or var1 var2...) or (not var1...)
+as in the common lisp reader, with OR being the default for multiple keys.
+Keys are space seperated.
+Keys may also consist of key=value pairs, where the key is checked for equalness
+to the value. Note that values may contain no spaces here. (TODO: support quotes
+in values)
 
-  #CONDITIONED_LINE(win32): $(SRC_DIR)/atomic/gcc_x86$(O)
+  #+(var1 var2...) defaults to #+(or var1 var2...)
+  #-(var1 var2...) defaults to #-(or var1 var2...)
+
+For instance:
+
+  #+(win32): $(SRC_DIR)/atomic/gcc_x86$(O)
 
 will be processed if the platform is win32.
 
+  #-(win32 cygwin): $(SRC_DIR)/atomic/gcc_x86$(O)
+
+will be skipped if the platform is win32 or cygwin.
+
+  #+(and win32 glut (not cygwin)):
+
+will be used on win32 and if glut is defined, but not on cygwin.
+
 =item comment_type
 
 This option takes has two possible values, C<#> or C</*>. If present and
@@ -329,12 +352,20 @@ sub genfile {
             last;
         }
         if ( $options{conditioned_lines} ) {
-            if ( $line =~ m/^#CONDITIONED_LINE\(([^)]+)\):(.*)/s ) {
-                next unless $conf->data->get($1);
+	    # allow nested parens here
+            if ( $line =~ m/^#([-+])\((.+)\):(.*)/s ) {
+		my $truth = cond_eval($conf, $2);
+		next if ($1 eq '-') and $truth;
+		next if ($1 eq '+') and not $truth;
+                $line = $3;
+	    }
+	    # but here not (legacy)
+	    elsif ( $line =~ m/^#CONDITIONED_LINE\(([^)]+)\):(.*)/s ) {
+                next unless cond_eval($conf, $1);
                 $line = $2;
             }
             elsif ( $line =~ m/^#INVERSE_CONDITIONED_LINE\(([^)]+)\):(.*)/s ) {
-                next if $conf->data->get($1);
+                next if cond_eval($conf, $1);
                 $line = $2;
             }
         }
@@ -434,6 +465,18 @@ sub genfile {
     move_if_diff( "$target.tmp", $target, $options{ignore_pattern} );
 }
 
+# Just checks the logical truth of the hash value (exists and not empty)
+# TODO: also check the platform name if the hash key does not exist
+# TODO: recursive (), evaluate AND, OR, NOT with multiple keys
+# TODO: check for key=value, like #+(ld=gcc)
+sub cond_eval {
+    my $conf = shift;
+    my $expr = shift;
+    # TODO: parse for parens and multiple keys in the expression and
+    # evaluate it recursively.
+    return $conf->data->get($expr);
+}
+
 sub append_configure_log {
     my $conf = shift;
     my $target = shift;
Index: parrot-svn/config/gen/makefiles/bcg.in
===================================================================
--- parrot-svn.orig/config/gen/makefiles/bcg.in
+++ parrot-svn/config/gen/makefiles/bcg.in
@@ -1,8 +1,8 @@
 RM_F		= @rm_f@
 PERL		= @perl@
 PARROT	    	= @build_dir@/parrot@exe@
-#CONDITIONED_LINE(win32):LIBPARROT = @build_dir@/libparrot@a@
-#INVERSE_CONDITIONED_LINE(win32):LIBPARROT = @build_dir@/@blib_dir@/libparrot@a@
+#-(win32):LIBPARROT = @build_dir@/libparrot@a@
+#-(win32):LIBPARROT = @build_dir@/@blib_dir@/libparrot@a@
 O		= @o@
 LOAD_EXT	= @load_ext@
 CC         	= @cc@
@@ -14,8 +14,8 @@ LINK_DYNAMIC  	= @link_dynamic@
 LINK       	= @link@
 LINKFLAGS  	= @linkflags@ @link_debug@ @ld_debug@
 LD         	= @ld@
-#CONDITIONED_LINE(win32):LDFLAGS = @ldflags@ @ld_debug@ $(LIBPARROT)
-#INVERSE_CONDITIONED_LINE(win32):LDFLAGS = @ldflags@ @ld_debug@ -L@build_dir@/blib/lib/ -lparrot
+#-(win32):LDFLAGS = @ldflags@ @ld_debug@ $(LIBPARROT)
+#-(win32):LDFLAGS = @ldflags@ @ld_debug@ -L@build_dir@/blib/lib/ -lparrot
 SHARE_EXT       = @share_ext@
 LD_SHARE_FLAGS  = @ld_share_flags@
 BUILD_PMC   	= $(PERL) @build_dir@/tools/build/dynpmc.pl
Index: parrot-svn/config/gen/makefiles/docs.in
===================================================================
--- parrot-svn.orig/config/gen/makefiles/docs.in
+++ parrot-svn/config/gen/makefiles/docs.in
@@ -41,8 +41,8 @@ doc-prep:
 	$(MKPATH) ops
 
 packfile-c.pod: ../src/packfile.c
-#CONDITIONED_LINE(new_perldoc):	perldoc -ud packfile-c.pod ../src/packfile.c
-#INVERSE_CONDITIONED_LINE(new_perldoc):	perldoc -u ../src/packfile.c > packfile-c.pod
+#-(new_perldoc):	perldoc -ud packfile-c.pod ../src/packfile.c
+#-(new_perldoc):	perldoc -u ../src/packfile.c > packfile-c.pod
 
 clean:
 	$(RM_F) packfile-c.pod $(POD)
Index: parrot-svn/config/gen/makefiles/dynoplibs_pl.in
===================================================================
--- parrot-svn.orig/config/gen/makefiles/dynoplibs_pl.in
+++ parrot-svn/config/gen/makefiles/dynoplibs_pl.in
@@ -32,7 +32,7 @@ our $O = q[@o@];
 our $CFLAGS = q[@ccflags@ @cc_shared@ @cc_debug@ @ccwarn@ @cc_hasjit@ @cg_flag@ @gc_flag@];
 
 our $LIBPARROT = q[];
-#CONDITIONED_LINE(parrot_is_shared):$LIBPARROT = q[@libparrot_ldflags@];
+#-(parrot_is_shared):$LIBPARROT = q[@libparrot_ldflags@];
 
 # Here comes some stuff for Win32.
 our $PATHQUOTE = q[];
@@ -53,8 +53,8 @@ our $OPS2C = "$PERL -I $PATHQUOTE" .
              $PATHQUOTE;
 our %cores = (
     C       => q[],
-#CONDITIONED_LINE(cg_flag):   CGP     => '_cgp',
-#CONDITIONED_LINE(cg_flag):   CGoto   => '_cg',
+#-(cg_flag):   CGP     => '_cgp',
+#-(cg_flag):   CGoto   => '_cg',
     CSwitch => '_switch'
 );
 
Index: parrot-svn/config/gen/makefiles/dynpmc.in
===================================================================
--- parrot-svn.orig/config/gen/makefiles/dynpmc.in
+++ parrot-svn/config/gen/makefiles/dynpmc.in
@@ -14,30 +14,30 @@ PMCS = \
   dynlexpad \
   foo \
   rotest \
-#CONDITIONED_LINE(has_gdbm):  gdbmhash \
+#-(has_gdbm):  gdbmhash \
   rational \
-#CONDITIONED_LINE(has_crypto):  md2 \
-#CONDITIONED_LINE(has_crypto):  md4 \
-#CONDITIONED_LINE(has_crypto):  md5 \
-#CONDITIONED_LINE(has_crypto):  ripemd160 \
-#CONDITIONED_LINE(has_crypto):  sha \
-#CONDITIONED_LINE(has_crypto):  sha1 \
-#CONDITIONED_LINE(has_crypto):  sha256 \
-#CONDITIONED_LINE(has_crypto):  sha512 \
+#-(has_crypto):  md2 \
+#-(has_crypto):  md4 \
+#-(has_crypto):  md5 \
+#-(has_crypto):  ripemd160 \
+#-(has_crypto):  sha \
+#-(has_crypto):  sha1 \
+#-(has_crypto):  sha256 \
+#-(has_crypto):  sha512 \
   subproxy 
 
 BUILD = $(PERL) @build_dir@/tools/build/dynpmc.pl
 
 # some *.pmc file are generated
 GEN_PMC_DEFINITIONS = \
-#CONDITIONED_LINE(has_crypto):  md2.pmc \
-#CONDITIONED_LINE(has_crypto):  md4.pmc \
-#CONDITIONED_LINE(has_crypto):  md5.pmc \
-#CONDITIONED_LINE(has_crypto):  ripemd160.pmc \
-#CONDITIONED_LINE(has_crypto):  sha.pmc \
-#CONDITIONED_LINE(has_crypto):  sha1.pmc \
-#CONDITIONED_LINE(has_crypto):  sha256.pmc \
-#CONDITIONED_LINE(has_crypto):  sha512.pmc
+#-(has_crypto):  md2.pmc \
+#-(has_crypto):  md4.pmc \
+#-(has_crypto):  md5.pmc \
+#-(has_crypto):  ripemd160.pmc \
+#-(has_crypto):  sha.pmc \
+#-(has_crypto):  sha1.pmc \
+#-(has_crypto):  sha256.pmc \
+#-(has_crypto):  sha512.pmc
 
 all :
 	@$(BUILD) generate $(PMCS)
Index: parrot-svn/config/gen/makefiles/dynpmc_pl.in
===================================================================
--- parrot-svn.orig/config/gen/makefiles/dynpmc_pl.in
+++ parrot-svn/config/gen/makefiles/dynpmc_pl.in
@@ -36,7 +36,7 @@ our $O = q[@o@];
 our $CFLAGS = q[@ccflags@ @cc_shared@ @cc_debug@ @ccwarn@ @cc_hasjit@ @cg_flag@ @gc_flag@];
 
 our $LIBPARROT = q[];
-#CONDITIONED_LINE(parrot_is_shared):$LIBPARROT = q[@libparrot_ldflags@];
+#-(parrot_is_shared):$LIBPARROT = q[@libparrot_ldflags@];
 
 # Here comes some stuff for Win32.
 our $PATHQUOTE = '';
Index: parrot-svn/config/gen/makefiles/ext.in
===================================================================
--- parrot-svn.orig/config/gen/makefiles/ext.in
+++ parrot-svn/config/gen/makefiles/ext.in
@@ -69,12 +69,12 @@ clean: \
 Parrot-Embed : Parrot-Embed.dummy
 
 Parrot-Embed.dummy:
-#CONDITIONED_LINE(win32):	- cd Parrot-Embed && $(PERL) Makefile.PL && nmake
-#INVERSE_CONDITIONED_LINE(win32):	- cd Parrot-Embed && $(PERL) Makefile.PL && $(MAKE)
+#-(win32):	- cd Parrot-Embed && $(PERL) Makefile.PL && nmake
+#-(win32):	- cd Parrot-Embed && $(PERL) Makefile.PL && $(MAKE)
 
 Parrot-Embed.clean:
-#CONDITIONED_LINE(win32):	- cd Parrot-Embed && nmake clean
-#INVERSE_CONDITIONED_LINE(win32):	- cd Parrot-Embed && $(MAKE) clean
+#-(win32):	- cd Parrot-Embed && nmake clean
+#-(win32):	- cd Parrot-Embed && $(MAKE) clean
 
 #
 # Local variables:
Index: parrot-svn/config/gen/makefiles/json.in
===================================================================
--- parrot-svn.orig/config/gen/makefiles/json.in
+++ parrot-svn/config/gen/makefiles/json.in
@@ -4,8 +4,8 @@ PERL     = @perl@
 RM_F     = @rm_f@
 PARROT   = ../../parrot@exe@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(cygchkdll):SHRPENV  = env PATH="@build_dir@/blib/lib:$(PATH)"
-#INVERSE_CONDITIONED_LINE(cygchkdll):SHRPENV  = env LD_RUN_PATH=@build_dir@/blib/lib
+#-(cygchkdll):SHRPENV  = env PATH="@build_dir@/blib/lib:$(PATH)"
+#-(cygchkdll):SHRPENV  = env LD_RUN_PATH=@build_dir@/blib/lib
 
 TOOL_DIR = ../..
 PGE_DIR  = ../../compilers/pge
Index: parrot-svn/config/gen/makefiles/nqp.in
===================================================================
--- parrot-svn.orig/config/gen/makefiles/nqp.in
+++ parrot-svn/config/gen/makefiles/nqp.in
@@ -5,8 +5,8 @@ PERL     = @perl@
 RM_F     = @rm_f@
 PARROT   = ../../parrot@exe@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(cygchkdll): SHRPENV  = env PATH="@build_dir@/blib/lib:$(PATH)"
-#INVERSE_CONDITIONED_LINE(cygchkdll): SHRPENV  = env LD_RUN_PATH=@build_dir@/blib/lib
+#-(cygchkdll): SHRPENV  = env PATH="@build_dir@/blib/lib:$(PATH)"
+#-(cygchkdll): SHRPENV  = env LD_RUN_PATH=@build_dir@/blib/lib
 
 PARROT_LIBRARY = @build_dir@/runtime/parrot/library
 PGE_LIBRARY  = @build_dir@/runtime/parrot/library/PGE
Index: parrot-svn/config/gen/makefiles/parrot_embed.in
===================================================================
--- parrot-svn.orig/config/gen/makefiles/parrot_embed.in
+++ parrot-svn/config/gen/makefiles/parrot_embed.in
@@ -15,7 +15,7 @@ $config{PARROTDIR}       = '../..';
 $config{PARROT}          = '..@slash@..@slash@parrot@exe@';
 $config{C_LIBS}          = '@libs@';
 $config{INCLUDE}         = "$config{PARROTDIR}/include";
-#INVERSE_CONDITIONED_LINE(win32):$config{ALL_PARROT_LIBS} = "@libparrot_ldflags@ $config{C_LIBS}";
+#-(win32):$config{ALL_PARROT_LIBS} = "@libparrot_ldflags@ $config{C_LIBS}";
 $config{ABS_PARROTDIR}   = Cwd::realpath(
                                File::Spec->rel2abs( $config{PARROTDIR} ) );
 $config{LDDLFLAGS}       = $Config{lddlflags};
@@ -30,9 +30,9 @@ WriteMakefile(
     'NAME'         => 'Parrot::Embed',
     'VERSION_FROM' => 'lib/Parrot/Embed.pm',
     'PREREQ_PM'    => { 'ExtUtils::CBuilder' => 0 },
-#CONDITIONED_LINE(win32):    'LIBS'         => [ $config{C_LIBS} ],
-#CONDITIONED_LINE(win32):    'OBJECT'       => "@libparrot_ldflags@ Embed@o@",
-#INVERSE_CONDITIONED_LINE(win32):    'LIBS'         => [ $config{ALL_PARROT_LIBS} ],
+#-(win32):    'LIBS'         => [ $config{C_LIBS} ],
+#-(win32):    'OBJECT'       => "@libparrot_ldflags@ Embed@o@",
+#-(win32):    'LIBS'         => [ $config{ALL_PARROT_LIBS} ],
     'INC'          => "-I$config{INCLUDE}",
     'PM'           => { map { $_ => "blib/$_" } <lib/Parrot/*pm> },
     'clean'        => { FILES => '*.xs t/greet.pbc' },
Index: parrot-svn/config/gen/makefiles/pct.in
===================================================================
--- parrot-svn.orig/config/gen/makefiles/pct.in
+++ parrot-svn/config/gen/makefiles/pct.in
@@ -10,8 +10,8 @@ PGE_DIR  = ../../compilers/pge
 TEST_DIR = ../../t/compilers/pct
 CP       = @cp@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(cygchkdll):SHRPENV  = env PATH="@build_dir@/blib/lib:$(PATH)"
-#INVERSE_CONDITIONED_LINE(cygchkdll):SHRPENV  = env LD_RUN_PATH=@build_dir@/blib/lib
+#-(cygchkdll):SHRPENV  = env PATH="@build_dir@/blib/lib:$(PATH)"
+#-(cygchkdll):SHRPENV  = env LD_RUN_PATH=@build_dir@/blib/lib
 
 # Where to put things
 PARROT_LIBRARY    = ../../runtime/parrot/library
Index: parrot-svn/config/gen/makefiles/pge.in
===================================================================
--- parrot-svn.orig/config/gen/makefiles/pge.in
+++ parrot-svn/config/gen/makefiles/pge.in
@@ -8,8 +8,8 @@ PARROT   = ../../parrot@exe@
 TOOL_DIR = ../..
 CP       = @cp@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(cygchkdll):SHRPENV  = env PATH="@build_dir@/blib/lib:$(PATH)"
-#INVERSE_CONDITIONED_LINE(cygchkdll):SHRPENV  = env LD_RUN_PATH=@build_dir@/blib/lib
+#-(cygchkdll):SHRPENV  = env PATH="@build_dir@/blib/lib:$(PATH)"
+#-(cygchkdll):SHRPENV  = env LD_RUN_PATH=@build_dir@/blib/lib
 
 # Where to put things
 PARROT_LIBRARY    = ../../runtime/parrot/library
Index: parrot-svn/config/gen/makefiles/tge.in
===================================================================
--- parrot-svn.orig/config/gen/makefiles/tge.in
+++ parrot-svn/config/gen/makefiles/tge.in
@@ -7,8 +7,8 @@ PERL     = @perl@
 RM_F     = @rm_f@
 PARROT   = ../../parrot@exe@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(cygchkdll):SHRPENV  = env PATH="@build_dir@/blib/lib:$(PATH)"
-#INVERSE_CONDITIONED_LINE(cygchkdll):SHRPENV  = env LD_RUN_PATH=@build_dir@/blib/lib
+#-(cygchkdll):SHRPENV  = env PATH="@build_dir@/blib/lib:$(PATH)"
+#-(cygchkdll):SHRPENV  = env LD_RUN_PATH=@build_dir@/blib/lib
 
 # Where to put things
 PARROT_LIBRARY    = ../../runtime/parrot/library
Index: parrot-svn/languages/APL/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/APL/config/makefiles/root.in
+++ parrot-svn/languages/APL/config/makefiles/root.in
@@ -24,9 +24,9 @@ PBC_TO_EXE    = ../../pbc_to_exe@exe@
 BUILD_DYNPMC  = $(PERL) $(BUILD_DIR)/tools/build/dynpmc.pl
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 ## places to look for things
 PARROT_DYNEXT = $(BUILD_DIR)/runtime/parrot/dynext
Index: parrot-svn/languages/PIR/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/PIR/config/makefiles/root.in
+++ parrot-svn/languages/PIR/config/makefiles/root.in
@@ -17,9 +17,9 @@ PARROT        = ../../parrot@exe@
 CAT           = $(PERL) -MExtUtils::Command -e cat
 BUILD_DYNPMC  = $(PERL) $(BUILD_DIR)/tools/build/dynpmc.pl
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 ## places to look for things
 PARROT_DYNEXT = $(BUILD_DIR)/runtime/parrot/dynext
Index: parrot-svn/languages/WMLScript/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/WMLScript/config/makefiles/root.in
+++ parrot-svn/languages/WMLScript/config/makefiles/root.in
@@ -21,9 +21,9 @@ MANDIR	      = @mandir@
 PARROT_DYNEXT = @build_dir@/runtime/parrot/dynext
 PBC_TO_EXE    = ../../pbc_to_exe@exe@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 SRM=Stack
 
Index: parrot-svn/languages/Zcode/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/Zcode/config/makefiles/root.in
+++ parrot-svn/languages/Zcode/config/makefiles/root.in
@@ -8,9 +8,9 @@ PARROT     = ../../parrot@exe@
 PERL       = @perl@
 RM_F       = @rm_f@
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 EXAMPLES = examples/small.z3 examples/hello.z3
 
Index: parrot-svn/languages/abc/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/abc/config/makefiles/root.in
+++ parrot-svn/languages/abc/config/makefiles/root.in
@@ -24,9 +24,9 @@ PBC_TO_EXE    = ../../pbc_to_exe@exe@
 BUILD_DYNPMC  = $(PERL) $(BUILD_DIR)/tools/build/dynpmc.pl
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 ## places to look for things
 PARROT_DYNEXT = $(BUILD_DIR)/runtime/parrot/dynext
Index: parrot-svn/languages/befunge/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/befunge/config/makefiles/root.in
+++ parrot-svn/languages/befunge/config/makefiles/root.in
@@ -14,9 +14,9 @@ PARROT     = ../../parrot
 PERL       = @perl@
 RM_F       = @rm_f@
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 DOCS = MAINTAINER README Changes
 
Index: parrot-svn/languages/bf/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/bf/config/makefiles/root.in
+++ parrot-svn/languages/bf/config/makefiles/root.in
@@ -9,9 +9,9 @@ RM_F = @rm_f@
 PERL = @perl@
 PARROT = ../../parrot@exe@
 PBC_TO_EXE    = ../../pbc_to_exe@exe@
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 DOCS = MAINTAINER README
 
Index: parrot-svn/languages/c99/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/c99/config/makefiles/root.in
+++ parrot-svn/languages/c99/config/makefiles/root.in
@@ -24,9 +24,9 @@ CAT           = $(PERL) -MExtUtils::Comm
 BUILD_DYNPMC  = $(PERL) $(BUILD_DIR)/tools/build/dynpmc.pl
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
 POD2MAN       = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 ## places to look for things
 PARROT_DYNEXT = $(BUILD_DIR)/runtime/parrot/dynext
Index: parrot-svn/languages/cardinal/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/cardinal/config/makefiles/root.in
+++ parrot-svn/languages/cardinal/config/makefiles/root.in
@@ -24,9 +24,9 @@ PBC_TO_EXE    = ../../pbc_to_exe@exe@
 BUILD_DYNPMC  = $(PERL) $(BUILD_DIR)/tools/build/dynpmc.pl
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 ## places to look for things
 PARROT_DYNEXT = $(BUILD_DIR)/runtime/parrot/dynext
Index: parrot-svn/languages/chitchat/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/chitchat/config/makefiles/root.in
+++ parrot-svn/languages/chitchat/config/makefiles/root.in
@@ -24,9 +24,9 @@ PBC_TO_EXE    = ../../pbc_to_exe@exe@
 BUILD_DYNPMC  = $(PERL) $(BUILD_DIR)/tools/build/dynpmc.pl
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 ## places to look for things
 PARROT_DYNEXT = $(BUILD_DIR)/runtime/parrot/dynext
Index: parrot-svn/languages/cola/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/cola/config/makefiles/root.in
+++ parrot-svn/languages/cola/config/makefiles/root.in
@@ -22,9 +22,9 @@ YACC          = @yacc@
 LEX           = @lex@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 # Libraries:  Ordinarily, no special libraries are needed
 # for cola, but one of the following settings might be helpful
Index: parrot-svn/languages/eclectus/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/eclectus/config/makefiles/root.in
+++ parrot-svn/languages/eclectus/config/makefiles/root.in
@@ -21,9 +21,9 @@ PBC_TO_EXE    = ../../pbc_to_exe@exe@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
 BUILD_DYNPMC  = $(PERL) @build_dir@/tools/build/dynpmc.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 # Set up directories
 PARROT_DYNEXT   = @build_dir@/runtime/parrot/dynext
Index: parrot-svn/languages/ecmascript/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/ecmascript/config/makefiles/root.in
+++ parrot-svn/languages/ecmascript/config/makefiles/root.in
@@ -25,9 +25,9 @@ PBC_TO_EXE    = ../../pbc_to_exe@exe@
 BUILD_DYNPMC  = $(PERL) $(BUILD_DIR)/tools/build/dynpmc.pl
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 ## places to look for things
 PARROT_DYNEXT = $(BUILD_DIR)/runtime/parrot/dynext
Index: parrot-svn/languages/forth/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/forth/config/makefiles/root.in
+++ parrot-svn/languages/forth/config/makefiles/root.in
@@ -8,9 +8,9 @@ PBC_MERGE     = ../../pbc_merge@exe@
 PBC_TO_EXE    = ../../pbc_to_exe@exe@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 BUILD_DIR     = @build_dir@
 BIN_DIR       = @bin_dir@
Index: parrot-svn/languages/hq9plus/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/hq9plus/config/makefiles/root.in
+++ parrot-svn/languages/hq9plus/config/makefiles/root.in
@@ -20,9 +20,9 @@ PBC_TO_EXE    = ../../pbc_to_exe@exe@
 CAT           = $(PERL) -MExtUtils::Command -e cat
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 ## places to look for things
 PGE_LIBRARY   = $(BUILD_DIR)/runtime/parrot/library/PGE
Index: parrot-svn/languages/jako/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/jako/config/makefiles/root.in
+++ parrot-svn/languages/jako/config/makefiles/root.in
@@ -17,9 +17,9 @@ INTERP        = ../../@test_prog@
 PBC_TO_EXE    = ../../pbc_to_exe@exe@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 .SUFFIXES: .jako .pir
 DOCS = README MAINTAINER
Index: parrot-svn/languages/json/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/json/config/makefiles/root.in
+++ parrot-svn/languages/json/config/makefiles/root.in
@@ -24,9 +24,9 @@ PBC_TO_EXE    = ../../pbc_to_exe@exe@
 BUILD_DYNPMC  = $(PERL) $(BUILD_DIR)/tools/build/dynpmc.pl
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 ## places to look for things
 PARROT_DYNEXT = $(BUILD_DIR)/runtime/parrot/dynext
Index: parrot-svn/languages/lazy-k/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/lazy-k/config/makefiles/root.in
+++ parrot-svn/languages/lazy-k/config/makefiles/root.in
@@ -10,9 +10,9 @@ RM_F          = @rm_f@
 PBC_TO_EXE    = ../../pbc_to_exe@exe@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 # set up location
 BUILD_DIR     = @build_dir@
Index: parrot-svn/languages/lisp/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/lisp/config/makefiles/root.in
+++ parrot-svn/languages/lisp/config/makefiles/root.in
@@ -24,9 +24,9 @@ PBC_TO_EXE    = ../../pbc_to_exe@exe@
 BUILD_DYNPMC  = $(PERL) $(BUILD_DIR)/tools/build/dynpmc.pl
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 DOCS = CHANGES KNOWN_BUGS LICENSE MAINTAINER README
 
Index: parrot-svn/languages/lolcode/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/lolcode/config/makefiles/root.in
+++ parrot-svn/languages/lolcode/config/makefiles/root.in
@@ -24,9 +24,9 @@ PBC_TO_EXE    = ../../pbc_to_exe@exe@
 BUILD_DYNPMC  = $(PERL) $(BUILD_DIR)/tools/build/dynpmc.pl
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 ## places to look for things
 PARROT_DYNEXT = $(BUILD_DIR)/runtime/parrot/dynext
Index: parrot-svn/languages/lua/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/lua/config/makefiles/root.in
+++ parrot-svn/languages/lua/config/makefiles/root.in
@@ -17,9 +17,9 @@ PGE         = $(PARROT) ../../runtime/pa
 TGE         = $(PARROT) ../../compilers/tge/tgc.pir
 RECONFIGURE = $(PERL) @build_dir@/tools/dev/reconfigure.pl
 PMCBUILD = $(PERL) @build_dir@/tools/build/dynpmc.pl
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 ## places to look for things
 LIBPATH       = src/lib
Index: parrot-svn/languages/m4/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/m4/config/makefiles/root.in
+++ parrot-svn/languages/m4/config/makefiles/root.in
@@ -16,9 +16,9 @@ MANDIR	      = @mandir@
 PBC_TO_EXE    = ../../pbc_to_exe@exe@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 DOCS = BUGS ChangeLog LICENSE MAINTAINER README TODO.pod VERSION \
        pod/features.pod pod/freezing.pod pod/operation.pod pod/running.pod
Index: parrot-svn/languages/ook/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/ook/config/makefiles/root.in
+++ parrot-svn/languages/ook/config/makefiles/root.in
@@ -10,9 +10,9 @@ PERL          = @perl@
 RM_F          = @rm_f@
 POD2MAN       = pod2man
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 # Set up directories
 BUILD_DIR     = @build_dir@
Index: parrot-svn/languages/parrot_compiler/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/parrot_compiler/config/makefiles/root.in
+++ parrot-svn/languages/parrot_compiler/config/makefiles/root.in
@@ -8,9 +8,9 @@ PERL          = @perl@
 RM_F          = @rm_f@
 PARROT        = ../../@test_prog@@exe@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 # Set up directories
 BUILD_DIR     = @build_dir@
Index: parrot-svn/languages/perl6/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/perl6/config/makefiles/root.in
+++ parrot-svn/languages/perl6/config/makefiles/root.in
@@ -35,9 +35,9 @@ CAT           = $(PERL) -MExtUtils::Comm
 BUILD_DYNPMC  = $(PERL) $(BUILD_DIR)/tools/build/dynpmc.pl
 BUILD_DYNOPS  = $(PERL) $(BUILD_DIR)/tools/build/dynoplibs.pl
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 DOCS = README MAINTAINER ROADMAP docs/compiler_overview.pod docs/glossary.pod docs/STATUS
 
 build: perl6.pbc Test.pir
Index: parrot-svn/languages/pheme/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/pheme/config/makefiles/root.in
+++ parrot-svn/languages/pheme/config/makefiles/root.in
@@ -19,9 +19,9 @@ TGE_DIR       = ../../compilers/tge
 PGE_LIBRARY   = $(BUILD_DIR)/runtime/parrot/library/PGE
 PERL6GRAMMAR  = $(PGE_LIBRARY)/Perl6Grammar.pbc
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 DOCS = MAINTAINER MANIFEST README TODO
 
Index: parrot-svn/languages/punie/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/punie/config/makefiles/root.in
+++ parrot-svn/languages/punie/config/makefiles/root.in
@@ -17,9 +17,9 @@ PGE_DIR       = ../../compilers/pge
 NQP_DIR       = ../../compilers/nqp
 CP            = @cp@
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 PARROT_LIBRARY = ../../runtime/parrot/library
 PGE_LIBRARY   = ../../runtime/parrot/library/PGE
Index: parrot-svn/languages/pynie/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/pynie/config/makefiles/root.in
+++ parrot-svn/languages/pynie/config/makefiles/root.in
@@ -24,9 +24,9 @@ CAT           = $(PERL) -MExtUtils::Comm
 BUILD_DYNPMC  = $(PERL) $(BUILD_DIR)/tools/build/dynpmc.pl
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 ## places to look for things
 PARROT_DYNEXT = $(BUILD_DIR)/runtime/parrot/dynext
Index: parrot-svn/languages/regex/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/regex/config/makefiles/root.in
+++ parrot-svn/languages/regex/config/makefiles/root.in
@@ -12,9 +12,9 @@ PMC_DIR       = pmc
 BUILD_DYNPMC  = $(PERL) @build_dir@/tools/build/dynpmc.pl
 PARROT_DYNEXT = @build_dir@/runtime/parrot/dynext
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 # Set up directories
 LOAD_EXT      = @load_ext@
Index: parrot-svn/languages/scheme/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/scheme/config/makefiles/root.in
+++ parrot-svn/languages/scheme/config/makefiles/root.in
@@ -8,9 +8,9 @@ LANG          = scheme
 PERL          = @perl@
 RM_F          = @rm_f@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 # Set up directories
 BUILD_DIR     = @build_dir@
Index: parrot-svn/languages/squaak/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/squaak/config/makefiles/root.in
+++ parrot-svn/languages/squaak/config/makefiles/root.in
@@ -24,9 +24,9 @@ CAT           = $(PERL) -MExtUtils::Comm
 BUILD_DYNPMC  = $(PERL) $(BUILD_DIR)/tools/build/dynpmc.pl
 RECONFIGURE   = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
 POD2MAN	      = pod2man
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 ## places to look for things
 PARROT_DYNEXT = $(BUILD_DIR)/runtime/parrot/dynext
Index: parrot-svn/languages/tcl/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/tcl/config/makefiles/root.in
+++ parrot-svn/languages/tcl/config/makefiles/root.in
@@ -9,9 +9,9 @@ PARROT      = $(BUILD_DIR)/parrot@exe@
 PBC_TO_EXE  = $(BUILD_DIR)/pbc_to_exe@exe@
 RECONFIGURE = $(PERL) $(BUILD_DIR)/tools/dev/reconfigure.pl
 PBC_TO_EXE    = $(BUILD_DIR)/pbc_to_exe@exe@
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 TCL_LIB   = library
 PMCBUILD  = $(PERL) @build_dir@/tools/build/dynpmc.pl
@@ -318,7 +318,7 @@ runtime/builtins.pir \
 "$(PMCDIR)/*_group.h" \
 "$(PMCDIR)/*$(LOAD_EXT)" \
 "$(PMCDIR)/*.dump" \
-#CONDITIONED_LINE(ld_parrot_exe_def):"$(PMCDIR)/*.def" \
+#+(ld_parrot_exe_def):"$(PMCDIR)/*.def" \
 "$(PMCDIR)/*.c" \
 "$(PMCDIR)/*$(O)" \
 "$(PMCDIR)/*.exp" \
Index: parrot-svn/languages/unlambda/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/unlambda/config/makefiles/root.in
+++ parrot-svn/languages/unlambda/config/makefiles/root.in
@@ -18,9 +18,9 @@ PERL            = @perl@
 RM_F            = @rm_f@
 PBC_TO_EXE    = ../../pbc_to_exe@exe@
 RECONFIGURE     = $(PERL) @build_dir@/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 # The default target
 default: build
Index: parrot-svn/languages/urm/config/makefiles/root.in
===================================================================
--- parrot-svn.orig/languages/urm/config/makefiles/root.in
+++ parrot-svn/languages/urm/config/makefiles/root.in
@@ -9,9 +9,9 @@ PERL          = @perl@
 PARROT        = ../../parrot@exe@
 BUILD_DIR     = @build_dir@
 RECONFIGURE   = $(PERL) @build_dir@/tools/dev/reconfigure.pl
-#CONDITIONED_LINE(darwin):
-#CONDITIONED_LINE(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
-#CONDITIONED_LINE(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
+#+(darwin):
+#+(darwin):# MACOSX_DEPLOYMENT_TARGET must be defined for OS X compilation/linking
+#+(darwin):export MACOSX_DEPLOYMENT_TARGET := @osx_version@
 
 default: all
 
Index: parrot-svn/config/gen/makefiles/docs.in~
===================================================================
--- parrot-svn.orig/config/gen/makefiles/docs.in~
+++ parrot-svn/config/gen/makefiles/docs.in~
@@ -41,8 +41,8 @@ doc-prep:
 	$(MKPATH) ops
 
 packfile-c.pod: ../src/packfile.c
-#CONDITIONED_LINE(new_perldoc):	perldoc -ud packfile-c.pod ../src/packfile.c 
-#INVERSE_CONDITIONED_LINE(new_perldoc):	perldoc -u ../src/packfile.c > packfile-c.pod
+#-(new_perldoc):	perldoc -ud packfile-c.pod ../src/packfile.c 
+#-(new_perldoc):	perldoc -u ../src/packfile.c > packfile-c.pod
 
 clean:
 	$(RM_F) packfile-c.pod $(POD)
