--- origsrc/xapian-omega-1.0.9/Makefile.am	2008-11-01 00:45:28.000000000 +0100
+++ src/xapian-omega-1.0.9/Makefile.am	2008-12-13 13:14:24.015625000 +0100
@@ -9,14 +9,49 @@
 INCLUDES = -I$(top_srcdir)/common
 
 EXTRA_DIST =\
+	xapian-omega.spec
+
+AM_CPPFLAGS = -DCONFIGFILE_SYSTEM=\"$(sysconfdir)/omega.conf\"
+
+pkglibbindir = $(libdir)/omega/bin
+pkglibbin_PROGRAMS = omega
+bin_PROGRAMS = omindex scriptindex
+dist_bin_SCRIPTS = dbi2omega htdig2omega mbox2omega
+
+check_PROGRAMS = htmlparsetest md5test
+TESTS = htmlparsetest$(EXEEXT) md5test$(EXEEXT)
+
+# Install documentation.
+dist_doc_DATA = \
+	docs/cgiparams.txt \
+	docs/omegascript.txt \
+	docs/overview.txt \
+	docs/quickstart.txt \
+	docs/scriptindex.txt \
+	docs/termprefixes.txt
+
+omegadatadir = $(datadir)/omega
+dist_omegadata_DATA = htdig2omega.script mbox2omega.script
+
+dist_sysconf_DATA = omega.conf
+
+templatesdir = $(localstatedir)/lib/omega/templates
+templatesincdir = $(localstatedir)/lib/omega/templates/inc
+webimagesdir = $(omegadatadir)/images
+
+dist_templates_DATA = \
 	templates/query \
 	templates/topterms \
 	templates/opensearch \
 	templates/godmode \
-	templates/xml \
+	templates/xml
+
+dist_templatesinc_DATA = \
 	templates/inc/anyalldropbox \
 	templates/inc/anyallradio \
-	templates/inc/toptermsjs \
+	templates/inc/toptermsjs
+
+dist_webimages_DATA =\
 	images/nextoff.png \
 	images/next.png \
 	images/page-10.png \
