diff -urN origsrc/xapian-omega-1.0.0/CYGWIN-PATCHES/README src/xapian-omega-1.0.0/CYGWIN-PATCHES/README
--- origsrc/xapian-omega-1.0.0/CYGWIN-PATCHES/README	1969-12-31 19:00:00.000000000 -0500
+++ src/xapian-omega-1.0.0/CYGWIN-PATCHES/README	2007-05-20 09:01:56.406250000 -0400
@@ -0,0 +1,101 @@
+xapian-omega
+-------------------------------
+xapian is a fulltext indexer and search library. 
+The Omega package contains the indexers and the 
+Web Interface via CGI.
+
+Runtime requirements: (versions given or later)
+  cygwin-1.5.x
+  libxapian14
+  unzip
+  antiword   (optional)
+  readpst    (optional)
+  unrtf      (optional)
+  catdoc     (optional)
+  pdftotext  (optional)
+  pod2text   (optional)
+
+Build requirements: (versions given or later)
+  libxapian-devel
+  cygwin-1.5.x
+  zlib-1.2.x
+  gcc-core
+  gcc-g++
+  libtool
+  binutils
+  make
+  bash
+  fileutils
+  sed
+  xapian
+
+Canonical homepage:
+  http://www.xapian.org/
+
+Canonical download:
+  http://www.xapian.org/download.php
+
+-------------------------------------------------------------------------------
+
+Build instructions:
+  If you use setup to install this src package, it will be
+  unpacked under /usr/src automatically.
+
+  cd /usr/src
+  tar xfvj xapian-omega-<VER>-src.tar.bz2
+  cygport ./xapian-omega-<VER>-<CYGREL>.cygport almostall
+
+This will create:
+  /usr/src/xapian-omega-<VER>-<CYGREL>.tar.bz2
+  /usr/src/xapian-omega-<VER>-<CYGREL>-src.tar.bz2
+
+Or use 'cygport xapian-omega-<VER>-<CYGREL>.cygport prep' to get 
+a patched source directory 
+
+-------------------------------------------------------------------------------
+
+Files included in the xapian-omega package:
+
+  /etc/omega.conf
+  /usr/bin/dbi2omega
+  /usr/bin/htdig2omega
+  /usr/bin/mbox2omega
+  /usr/bin/omindex.exe
+  /usr/bin/scriptindex.exe
+  /usr/lib/omega/bin/omega.exe
+  /usr/share/doc/Cygwin/xapian-omega-1.0.0.README
+  /usr/share/doc/xapian-omega-1.0.0/AUTHORS
+  /usr/share/doc/xapian-omega-1.0.0/COPYING
+  /usr/share/doc/xapian-omega-1.0.0/ChangeLog
+  /usr/share/doc/xapian-omega-1.0.0/INSTALL
+  /usr/share/doc/xapian-omega-1.0.0/NEWS
+  /usr/share/doc/xapian-omega-1.0.0/README
+  /usr/share/doc/xapian-omega-1.0.0/TODO
+  /usr/share/doc/xapian-omega-1.0.0/cgiparams.txt
+  /usr/share/doc/xapian-omega-1.0.0/omegascript.txt
+  /usr/share/doc/xapian-omega-1.0.0/overview.txt
+  /usr/share/doc/xapian-omega-1.0.0/quickstart.txt
+  /usr/share/doc/xapian-omega-1.0.0/scriptindex.txt
+  /usr/share/doc/xapian-omega-1.0.0/termprefixes.txt
+  /usr/share/man/man1/omindex.1.gz
+  /usr/share/man/man1/scriptindex.1.gz
+  /usr/share/omega/htdig2omega.script
+  /usr/share/omega/mbox2omega.script
+
+-------------------------------------------------------------------------------
+
+Port Notes:
+
+----- version 1.0.0-1 -----
+require libxapian14
+----- version 0.9.9-1 -----
+require libxapian13
+----- version 0.9.8-1 -----
+no changes
+----- version 0.9.7-1 -----
+no changes
+----- version 0.9.6-1 ----- 
+Initial release
+
+Cygwin port maintained by: Reini Urban <rurban@x-ray.at>
+Cygwin port questions to the mailinglist <cygwin@cygwin.com>
diff -urN origsrc/xapian-omega-1.0.0/CYGWIN-PATCHES/xapian-omega.hint src/xapian-omega-1.0.0/CYGWIN-PATCHES/xapian-omega.hint
--- origsrc/xapian-omega-1.0.0/CYGWIN-PATCHES/xapian-omega.hint	1969-12-31 19:00:00.000000000 -0500
+++ src/xapian-omega-1.0.0/CYGWIN-PATCHES/xapian-omega.hint	2007-05-20 09:00:57.828125000 -0400
@@ -0,0 +1,6 @@
+sdesc: "Xapian Indexer and Web Interface (CGI)"
+ldesc: "Two index building utilities (omindex and scriptindex)
+and a CGI search application (omega).  These use the Xapian search library to
+provide a search over a collection of documents."
+category: Web
+requires: xapian libxapian14
