diff -urN ctris-0.41-orig/CYGWIN-PATCHES/setup.hint ctris-0.41/CYGWIN-PATCHES/setup.hint
--- ctris-0.41-orig/CYGWIN-PATCHES/setup.hint	1970-01-01 00:00:00.000000000 +0000
+++ ctris-0.41/CYGWIN-PATCHES/setup.hint	2006-07-19 06:11:40.140625000 +0000
@@ -0,0 +1,6 @@
+sdesc: "Console mode tetris clone"
+ldesc: "ctris is a tetris clone, formerly named ctetris. 
+It works very fast and efficient, so it should be possible to play it even over a 
+slow remote terminal. With highscores. License: GPL"
+category: Games
+requires: libncurses7 cygwin
